// @abp/ng.theme.lepton-x/esm2020/layouts/side-menu-layout/side-menu-layout.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';
import { CoreModule } from '@abp/ng.core';
import { LpxSideMenuLayoutModule } from '@volo/ngx-lepton-x.lite/layouts';
import { LpxFooterModule, LpxNavbarModule, LpxTranslateModule, PanelsModule, } from '@volo/ngx-lepton-x.core';
import { LanguageSelectionModule, UserProfileModule, } from '@volo/ngx-lepton-x.lite';
import { SideMenuApplicationLayoutComponent } from './side-menu-application-layout/side-menu-application-layout.component';
import { LPX_LAYOUT_PROVIDER } from './providers/layout.provider';
import { LPX_NAVBAR_ITEMS_PROVIDER } from './providers/navbar-items.provider';
import { NAV_ITEM_PROVIDER } from './providers/nav-item.provider';
import { NavItemsComponent } from './components/nav-items/nav-items.component';
import { LanguageSelectionComponent } from './components/language-selection/language-selection.component';
import { UserProfileComponent } from './components/user-profile/user-profile.component';
import { NavigateToLoginComponent } from './components/navigate-to-login/navigate-to-login.component';
import { LeptonXAbpCoreModule } from '@volo/abp.ng.lepton-x.core';
import * as i0 from "@angular/core";
export class SideMenuLayoutModule {
    static forRoot() {
        return {
            ngModule: SideMenuLayoutModule,
            providers: [
                LPX_LAYOUT_PROVIDER,
                LPX_NAVBAR_ITEMS_PROVIDER,
                NAV_ITEM_PROVIDER,
                LpxSideMenuLayoutModule.forRoot().providers,
            ],
        };
    }
}
SideMenuLayoutModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: SideMenuLayoutModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
SideMenuLayoutModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.2.0", ngImport: i0, type: SideMenuLayoutModule, declarations: [SideMenuApplicationLayoutComponent,
        NavItemsComponent,
        LanguageSelectionComponent,
        UserProfileComponent,
        NavigateToLoginComponent], imports: [CommonModule,
        LpxSideMenuLayoutModule,
        RouterModule,
        LpxNavbarModule,
        CoreModule,
        LeptonXAbpCoreModule,
        PanelsModule,
        UserProfileModule,
        LanguageSelectionModule,
        LpxTranslateModule,
        LpxFooterModule] });
SideMenuLayoutModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: SideMenuLayoutModule, imports: [CommonModule,
        LpxSideMenuLayoutModule,
        RouterModule,
        LpxNavbarModule,
        CoreModule,
        LeptonXAbpCoreModule,
        PanelsModule,
        UserProfileModule,
        LanguageSelectionModule,
        LpxTranslateModule,
        LpxFooterModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: SideMenuLayoutModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        SideMenuApplicationLayoutComponent,
                        NavItemsComponent,
                        LanguageSelectionComponent,
                        UserProfileComponent,
                        NavigateToLoginComponent,
                    ],
                    imports: [
                        CommonModule,
                        LpxSideMenuLayoutModule,
                        RouterModule,
                        LpxNavbarModule,
                        CoreModule,
                        LeptonXAbpCoreModule,
                        PanelsModule,
                        UserProfileModule,
                        LanguageSelectionModule,
                        LpxTranslateModule,
                        LpxFooterModule,
                    ],
                }]
        }] });
