// @abp/ng.theme.shared/esm2020/extensions/lib/models/form-props.mjs
import { Prop, PropData, PropList, Props, PropsFactory, } from './props';
export class FormPropList extends PropList {
}
export class FormProps extends Props {
    constructor() {
        super(...arguments);
        this._ctor = FormPropList;
    }
}
export class GroupedFormPropList {
    constructor() {
        this.items = [];
    }
    addItem(item) {
        const groupName = item.group?.name;
        let group = this.items.find(i => i.group?.name === groupName);
        if (group) {
            group.formPropList.addTail(item);
        }
        else {
            group = {
                formPropList: new FormPropList(),
                group: item.group,
            };
            group.formPropList.addHead(item);
            this.items.push(group);
        }
    }
}
export class CreateFormPropsFactory extends PropsFactory {
    constructor() {
        super(...arguments);
        this._ctor = FormProps;
    }
}
export class EditFormPropsFactory extends PropsFactory {
    constructor() {
        super(...arguments);
        this._ctor = FormProps;
    }
}
export class FormProp extends Prop {
    constructor(options) {
        super(options.type, options.name, options.displayName, options.permission, options.visible, options.isExtra, options.template, options.className);
        this.group = options.group;
        this.className = options.className;
        this.asyncValidators = options.asyncValidators || (_ => []);
        this.validators = options.validators || (_ => []);
        this.disabled = options.disabled || (_ => false);
        this.readonly = options.readonly || (_ => false);
        this.autocomplete = options.autocomplete || 'off';
        this.options = options.options;
        this.id = options.id || options.name;
        const defaultValue = options.defaultValue;
        this.defaultValue = isFalsyValue(defaultValue) ? defaultValue : defaultValue || null;
        this.displayTextResolver = options.displayTextResolver;
    }
    static create(options) {
        return new FormProp(options);
    }
    static createMany(arrayOfOptions) {
        return arrayOfOptions.map(FormProp.create);
    }
}
export class FormPropData extends PropData {
    constructor(injector, record) {
        super();
        this.record = record;
        this.getInjected = injector.get.bind(injector);
    }
}
function isFalsyValue(defaultValue) {
    return [0, '', false].indexOf(defaultValue) > -1;
}
