// @abp/ng.theme.shared/esm2020/extensions/lib/models/props.mjs
import { LinkedList } from '@abp/utils';
export class PropList extends LinkedList {
}
export class PropData {
    get data() {
        return {
            getInjected: this.getInjected,
            index: this.index,
            record: this.record,
        };
    }
}
export class Prop {
    constructor(type, name, displayName, permission, visible = _ => true, isExtra = false, template, className, displayTextResolver) {
        this.type = type;
        this.name = name;
        this.displayName = displayName;
        this.permission = permission;
        this.visible = visible;
        this.isExtra = isExtra;
        this.template = template;
        this.className = className;
        this.displayTextResolver = displayTextResolver;
        this.displayName = this.displayName || this.name;
    }
}
export class PropsFactory {
    constructor() {
        this.contributorCallbacks = {};
    }
    get(name) {
        this.contributorCallbacks[name] = this.contributorCallbacks[name] || [];
        return new this._ctor(this.contributorCallbacks[name]);
    }
}
export class Props {
    constructor(callbackList) {
        this.callbackList = callbackList;
    }
    get props() {
        const propList = new this._ctor();
        this.callbackList.forEach(callback => callback(propList));
        return propList;
    }
    addContributor(contributeCallback) {
        this.callbackList.push(contributeCallback);
    }
    clearContributors() {
        while (this.callbackList.length)
            this.callbackList.pop();
    }
}
