// @abp/ng.theme.lepton-x/fesm2015/abp-ng.theme.lepton-x-layouts.mjs
import * as i0 from '@angular/core';
import { Component, APP_INITIALIZER, Injector, NgModule } from '@angular/core';
import * as i1 from '@abp/ng.core';
import { ReplaceableComponentsService, CoreModule } from '@abp/ng.core';
import * as i2 from '@volo/ngx-lepton-x.lite';
import { UserProfileModule, LanguageSelectionModule } from '@volo/ngx-lepton-x.lite';
import * as i1$1 from '@abp/ng.theme.shared';
import { NavItemsService } from '@abp/ng.theme.shared';
import * as i2$1 from '@angular/common';
import { CommonModule } from '@angular/common';
import * as i3 from '@volo/ngx-lepton-x.core';
import { LpxNavbarModule, PanelsModule, LpxTranslateModule, LpxFooterModule } from '@volo/ngx-lepton-x.core';
import * as i1$2 from '@volo/ngx-lepton-x.lite/layouts';
import { LpxSideMenuLayoutModule } from '@volo/ngx-lepton-x.lite/layouts';
import * as i2$2 from '@angular/router';
import { RouterModule } from '@angular/router';
import * as i5 from '@volo/abp.ng.lepton-x.core';
import { AbpNavbarService, LeptonXAbpCoreModule } from '@volo/abp.ng.lepton-x.core';
import { OAuthService } from 'angular-oauth2-oidc';

class LanguageSelectionComponent {
    constructor() {
        this.languageSelectionKey = "Theme.LanguagesComponent" /* eThemeLeptonXComponents.Languages */;
    }
}
LanguageSelectionComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: LanguageSelectionComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
LanguageSelectionComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "14.2.0", type: LanguageSelectionComponent, selector: "abp-language-selection", ngImport: i0, template: "<lpx-language-selection *abpReplaceableTemplate=\"{\r\n        componentKey: languageSelectionKey\r\n      }\">\r\n</lpx-language-selection>\r\n", dependencies: [{ kind: "directive", type: i1.ReplaceableTemplateDirective, selector: "[abpReplaceableTemplate]", inputs: ["abpReplaceableTemplate"] }, { kind: "component", type: i2.LanguageSelectionComponent, selector: "lpx-language-selection" }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: LanguageSelectionComponent, decorators: [{
            type: Component,
            args: [{ selector: 'abp-language-selection', template: "<lpx-language-selection *abpReplaceableTemplate=\"{\r\n        componentKey: languageSelectionKey\r\n      }\">\r\n</lpx-language-selection>\r\n" }]
        }] });

class NavItemsComponent {
    constructor(navItems) {
        this.navItems = navItems;
        this.trackByFn = (_, element) => element.id;
    }
}
NavItemsComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: NavItemsComponent, deps: [{ token: i1$1.NavItemsService }], target: i0.ɵɵFactoryTarget.Component });
NavItemsComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "14.2.0", type: NavItemsComponent, selector: "abp-nav-items", ngImport: i0, template: "<ng-container *ngFor=\"let item of navItems.items$ | async; trackBy: trackByFn\">\r\n  <!-- TODO: muhammaed: use pipe or etc instead of func exec -->\r\n  <ng-container *ngIf=\"item.visible()\">\r\n    <ng-container *abpPermission=\"item.requiredPolicy\">\r\n      <ng-container\r\n        *ngIf=\"item.component; else htmlTemplate\"\r\n        [ngComponentOutlet]=\"item.component\"\r\n      ></ng-container>\r\n      <ng-template #htmlTemplate>\r\n        <div\r\n          [innerHTML]=\"item.html\"\r\n          (click)=\"item.action ? item.action() : null\"\r\n        ></div>\r\n      </ng-template>\r\n    </ng-container>\r\n  </ng-container>\r\n</ng-container>\r\n", styles: [":host{all:inherit}\n"], dependencies: [{ kind: "directive", type: i2$1.NgComponentOutlet, selector: "[ngComponentOutlet]", inputs: ["ngComponentOutlet", "ngComponentOutletInjector", "ngComponentOutletContent", "ngComponentOutletNgModule", "ngComponentOutletNgModuleFactory"] }, { kind: "directive", type: i2$1.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }, { kind: "directive", type: i2$1.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { kind: "directive", type: i1.PermissionDirective, selector: "[abpPermission]", inputs: ["abpPermission", "abpPermissionRunChangeDetection"] }, { kind: "pipe", type: i2$1.AsyncPipe, name: "async" }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: NavItemsComponent, decorators: [{
            type: Component,
            args: [{ selector: 'abp-nav-items', template: "<ng-container *ngFor=\"let item of navItems.items$ | async; trackBy: trackByFn\">\r\n  <!-- TODO: muhammaed: use pipe or etc instead of func exec -->\r\n  <ng-container *ngIf=\"item.visible()\">\r\n    <ng-container *abpPermission=\"item.requiredPolicy\">\r\n      <ng-container\r\n        *ngIf=\"item.component; else htmlTemplate\"\r\n        [ngComponentOutlet]=\"item.component\"\r\n      ></ng-container>\r\n      <ng-template #htmlTemplate>\r\n        <div\r\n          [innerHTML]=\"item.html\"\r\n          (click)=\"item.action ? item.action() : null\"\r\n        ></div>\r\n      </ng-template>\r\n    </ng-container>\r\n  </ng-container>\r\n</ng-container>\r\n", styles: [":host{all:inherit}\n"] }]
        }], ctorParameters: function () { return [{ type: i1$1.NavItemsService }]; } });

class UserProfileComponent {
    constructor() {
        this.key = "Theme.CurrentUserComponent" /* eThemeLeptonXComponents.CurrentUser */;
    }
}
UserProfileComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: UserProfileComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
UserProfileComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "14.2.0", type: UserProfileComponent, selector: "abp-user-profile", ngImport: i0, template: "<lpx-user-profile *abpReplaceableTemplate=\"{componentKey:key}\"></lpx-user-profile>\r\n", dependencies: [{ kind: "directive", type: i1.ReplaceableTemplateDirective, selector: "[abpReplaceableTemplate]", inputs: ["abpReplaceableTemplate"] }, { kind: "component", type: i2.UserProfileComponent, selector: "lpx-user-profile" }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: UserProfileComponent, decorators: [{
            type: Component,
            args: [{ selector: 'abp-user-profile', template: "<lpx-user-profile *abpReplaceableTemplate=\"{componentKey:key}\"></lpx-user-profile>\r\n" }]
        }] });

class NavigateToLoginComponent {
    constructor(authService) {
        this.authService = authService;
    }
    navigateToLogin(event) {
        event.preventDefault();
        this.authService.navigateToLogin();
    }
}
NavigateToLoginComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: NavigateToLoginComponent, deps: [{ token: i1.AuthService }], target: i0.ɵɵFactoryTarget.Component });
NavigateToLoginComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "14.2.0", type: NavigateToLoginComponent, selector: "abp-navigate-to-login", ngImport: i0, template: `
    <a href="#" class="nav-link" (click)="navigateToLogin($event)">
      <span>{{ 'AbpUi::Login' | lpxTranslate | async }}</span>
    </a>
  `, isInline: true, dependencies: [{ kind: "pipe", type: i2$1.AsyncPipe, name: "async" }, { kind: "pipe", type: i3.TranslatePipe, name: "lpxTranslate" }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: NavigateToLoginComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'abp-navigate-to-login',
                    template: `
    <a href="#" class="nav-link" (click)="navigateToLogin($event)">
      <span>{{ 'AbpUi::Login' | lpxTranslate | async }}</span>
    </a>
  `,
                }]
        }], ctorParameters: function () { return [{ type: i1.AuthService }]; } });

class SideMenuApplicationLayoutComponent {
    constructor() {
        this.toolbarKey = "Theme.NavbarComponent" /* eThemeLeptonXComponents.Navbar */;
        this.navbarKey = "Theme.Sidebar" /* eThemeLeptonXComponents.Sidebar */;
        this.routesKey = "Theme.RoutesComponent" /* eThemeLeptonXComponents.Routes */;
        this.toolbarItemsKey = "Theme.NavItemsComponent" /* eThemeLeptonXComponents.NavItems */;
        this.breadcrumbKey = "Theme.BreadcrumbComponent" /* eThemeLeptonXComponents.Breadcrumb */;
        this.footerKey = "Theme.FooterComponent" /* eThemeLeptonXComponents.Footer */;
        this.mobileNavbarKey = "Theme.MobileNavbarComponent" /* eThemeLeptonXComponents.MobileNavbar */;
        this.mobileUserProfile = "Theme.MobileUserProfileComponent" /* eThemeLeptonXComponents.MobileUserProfile */;
        this.mobileLanguageSelection = "Theme.MobileLanguageSelectionComponent" /* eThemeLeptonXComponents.MobileLanguageSelection */;
        this.pageAlertContainerKey = "Theme.PageAlertContainerComponent" /* eThemeLeptonXComponents.PageAlertContainer */;
    }
}
SideMenuApplicationLayoutComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: SideMenuApplicationLayoutComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
SideMenuApplicationLayoutComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "14.2.0", type: SideMenuApplicationLayoutComponent, selector: "abp-application-layout", ngImport: i0, template: "<lpx-layout #layout>\r\n  <abp-page-alert-container\r\n    *abpReplaceableTemplate=\"{ componentKey: pageAlertContainerKey }\"\r\n  ></abp-page-alert-container>\r\n  <ng-template lpx-navbar-panel>\r\n    <lpx-navbar\r\n      *abpReplaceableTemplate=\"{\r\n        componentKey: navbarKey\r\n      }\"\r\n    >\r\n      <ng-template lpx-navbar-routes let-items>\r\n        <lpx-navbar-routes\r\n          *abpReplaceableTemplate=\"{\r\n            componentKey: routesKey,\r\n            inputs: {\r\n              navbarItems: items\r\n            }\r\n          }\"\r\n          [navbarItems]=\"items\"\r\n        >\r\n        </lpx-navbar-routes>\r\n      </ng-template>\r\n    </lpx-navbar>\r\n  </ng-template>\r\n  <ng-template lpx-topbar-content>\r\n    <abp-nav-items\r\n      *abpReplaceableTemplate=\"{\r\n        componentKey: toolbarItemsKey\r\n      }\"\r\n    ></abp-nav-items>\r\n  </ng-template>\r\n  <ng-container *abpIfReplaceableTemplateExists=\"breadcrumbKey\">\r\n    <ng-template lpx-breadcrumb-panel>\r\n      <ng-container\r\n        *abpReplaceableTemplate=\"{\r\n          componentKey: breadcrumbKey\r\n        }\"\r\n      >\r\n      </ng-container>\r\n    </ng-template>\r\n  </ng-container>\r\n\r\n  <ng-container *abpIfReplaceableTemplateExists=\"mobileNavbarKey\">\r\n    <ng-template lpx-mobile-navbar-panel>\r\n      <ng-container *abpReplaceableTemplate=\"{ componentKey: mobileNavbarKey }\">\r\n      </ng-container>\r\n    </ng-template>\r\n  </ng-container>\r\n\r\n  <ng-container *abpIfReplaceableTemplateExists=\"mobileUserProfile\">\r\n    <ng-template lpx-mn-user-profile-panel>\r\n      <ng-container\r\n        *abpReplaceableTemplate=\"{ componentKey: mobileUserProfile }\"\r\n      >\r\n      </ng-container>\r\n    </ng-template>\r\n  </ng-container>\r\n\r\n  <ng-template lpx-footer-panel>\r\n    <lpx-footer\r\n      *abpReplaceableTemplate=\"{\r\n        componentKey: footerKey\r\n      }\"\r\n    >\r\n    </lpx-footer>\r\n  </ng-template>\r\n\r\n  <ng-container *abpIfReplaceableTemplateExists=\"mobileLanguageSelection\">\r\n    <ng-template lpx-mn-language-selection>\r\n      <ng-container\r\n        *abpReplaceableTemplate=\"{ componentKey: mobileLanguageSelection }\"\r\n      >\r\n      </ng-container>\r\n    </ng-template>\r\n  </ng-container>\r\n  <router-outlet></router-outlet>\r\n</lpx-layout>\r\n", dependencies: [{ kind: "component", type: i1$2.SideMenuLayoutComponent, selector: "lpx-layout" }, { kind: "directive", type: i1$2.TopbarContentPanelDirective, selector: "ng-template[lpx-topbar-content]" }, { kind: "directive", type: i1$2.MobileUserProfilePanelDirective, selector: "ng-template[lpx-mn-user-profile-panel]" }, { kind: "directive", type: i1$2.MobileLanguageSelectionDirective, selector: "ng-template[lpx-mn-language-selection]" }, { kind: "directive", type: i2$2.RouterOutlet, selector: "router-outlet", outputs: ["activate", "deactivate", "attach", "detach"], exportAs: ["outlet"] }, { kind: "component", type: i3.NavbarComponent, selector: "lpx-navbar" }, { kind: "component", type: i3.NavbarRoutesComponent, selector: "lpx-navbar-routes", inputs: ["navbarItems", "routerItem"], outputs: ["routeClick"] }, { kind: "directive", type: i3.NavbarRoutesDirective, selector: "[lpx-navbar-routes],[lpxNavbarRoutes]", exportAs: ["lpxNavbarRoutes"] }, { kind: "directive", type: i1.ReplaceableTemplateDirective, selector: "[abpReplaceableTemplate]", inputs: ["abpReplaceableTemplate"] }, { kind: "directive", type: i5.IfReplaceableTemplateExistsDirective, selector: "[abpIfReplaceableTemplateExists]", inputs: ["abpIfReplaceableTemplateExists"] }, { kind: "component", type: i5.PageAlertContainerComponent, selector: "abp-page-alert-container" }, { kind: "directive", type: i3.BreadcrumbPanelDirective, selector: "ng-template[lpx-breadcrumb-panel]" }, { kind: "directive", type: i3.NavbarPanelDirective, selector: "ng-template[lpx-navbar-panel]" }, { kind: "directive", type: i3.FooterPanelDirective, selector: "ng-template[lpx-footer-panel]" }, { kind: "directive", type: i3.MobileNavbarPanelDirective, selector: "ng-template[lpx-mobile-navbar-panel]" }, { kind: "component", type: i3.FooterComponent, selector: "lpx-footer" }, { kind: "component", type: NavItemsComponent, selector: "abp-nav-items" }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: SideMenuApplicationLayoutComponent, decorators: [{
            type: Component,
            args: [{ selector: 'abp-application-layout', template: "<lpx-layout #layout>\r\n  <abp-page-alert-container\r\n    *abpReplaceableTemplate=\"{ componentKey: pageAlertContainerKey }\"\r\n  ></abp-page-alert-container>\r\n  <ng-template lpx-navbar-panel>\r\n    <lpx-navbar\r\n      *abpReplaceableTemplate=\"{\r\n        componentKey: navbarKey\r\n      }\"\r\n    >\r\n      <ng-template lpx-navbar-routes let-items>\r\n        <lpx-navbar-routes\r\n          *abpReplaceableTemplate=\"{\r\n            componentKey: routesKey,\r\n            inputs: {\r\n              navbarItems: items\r\n            }\r\n          }\"\r\n          [navbarItems]=\"items\"\r\n        >\r\n        </lpx-navbar-routes>\r\n      </ng-template>\r\n    </lpx-navbar>\r\n  </ng-template>\r\n  <ng-template lpx-topbar-content>\r\n    <abp-nav-items\r\n      *abpReplaceableTemplate=\"{\r\n        componentKey: toolbarItemsKey\r\n      }\"\r\n    ></abp-nav-items>\r\n  </ng-template>\r\n  <ng-container *abpIfReplaceableTemplateExists=\"breadcrumbKey\">\r\n    <ng-template lpx-breadcrumb-panel>\r\n      <ng-container\r\n        *abpReplaceableTemplate=\"{\r\n          componentKey: breadcrumbKey\r\n        }\"\r\n      >\r\n      </ng-container>\r\n    </ng-template>\r\n  </ng-container>\r\n\r\n  <ng-container *abpIfReplaceableTemplateExists=\"mobileNavbarKey\">\r\n    <ng-template lpx-mobile-navbar-panel>\r\n      <ng-container *abpReplaceableTemplate=\"{ componentKey: mobileNavbarKey }\">\r\n      </ng-container>\r\n    </ng-template>\r\n  </ng-container>\r\n\r\n  <ng-container *abpIfReplaceableTemplateExists=\"mobileUserProfile\">\r\n    <ng-template lpx-mn-user-profile-panel>\r\n      <ng-container\r\n        *abpReplaceableTemplate=\"{ componentKey: mobileUserProfile }\"\r\n      >\r\n      </ng-container>\r\n    </ng-template>\r\n  </ng-container>\r\n\r\n  <ng-template lpx-footer-panel>\r\n    <lpx-footer\r\n      *abpReplaceableTemplate=\"{\r\n        componentKey: footerKey\r\n      }\"\r\n    >\r\n    </lpx-footer>\r\n  </ng-template>\r\n\r\n  <ng-container *abpIfReplaceableTemplateExists=\"mobileLanguageSelection\">\r\n    <ng-template lpx-mn-language-selection>\r\n      <ng-container\r\n        *abpReplaceableTemplate=\"{ componentKey: mobileLanguageSelection }\"\r\n      >\r\n      </ng-container>\r\n    </ng-template>\r\n  </ng-container>\r\n  <router-outlet></router-outlet>\r\n</lpx-layout>\r\n" }]
        }] });

const LPX_LAYOUT_PROVIDER = {
    provide: APP_INITIALIZER,
    useFactory: initLayouts,
    deps: [Injector],
    multi: true,
};
function initLayouts(injector) {
    const replaceableComponents = injector.get(ReplaceableComponentsService);
    return () => {
        replaceableComponents.add({
            key: "Theme.ApplicationLayoutComponent" /* eThemeLeptonXComponents.ApplicationLayout */,
            component: SideMenuApplicationLayoutComponent,
        });
    };
}

const NAV_ITEM_PROVIDER = {
    provide: APP_INITIALIZER,
    multi: true,
    useFactory: addNavItems,
    deps: [Injector],
};
function addNavItems(injector) {
    return () => {
        const navItems = injector.get(NavItemsService);
        const oAuthService = injector.get(OAuthService);
        navItems.addItems([
            {
                id: "Theme.LoginComponent" /* eThemeLeptonXComponents.Login */,
                order: 100,
                visible: () => !oAuthService.hasValidAccessToken(),
                component: NavigateToLoginComponent,
            },
            {
                id: "Theme.LanguagesComponent" /* eThemeLeptonXComponents.Languages */,
                order: 100,
                component: LanguageSelectionComponent,
            },
            {
                id: "Theme.CurrentUserComponent" /* eThemeLeptonXComponents.CurrentUser */,
                order: 100,
                component: UserProfileComponent,
            },
        ]);
    };
}

const LPX_NAVBAR_ITEMS_PROVIDER = {
    provide: APP_INITIALIZER,
    multi: true,
    useFactory: initNavbarItems,
    deps: [Injector],
};
function initNavbarItems(injector) {
    const navbarService = injector.get(AbpNavbarService);
    return () => {
        navbarService.initRoutes();
    };
}

class SideMenuLayoutModule {
    static forRoot() {
        return {
            ngModule: SideMenuLayoutModule,
            providers: [
                LPX_LAYOUT_PROVIDER,
                LPX_NAVBAR_ITEMS_PROVIDER,
                NAV_ITEM_PROVIDER,
                LpxSideMenuLayoutModule.forRoot().providers,
            ],
        };
    }
}
SideMenuLayoutModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: SideMenuLayoutModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
SideMenuLayoutModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.2.0", ngImport: i0, type: SideMenuLayoutModule, declarations: [SideMenuApplicationLayoutComponent,
        NavItemsComponent,
        LanguageSelectionComponent,
        UserProfileComponent,
        NavigateToLoginComponent], imports: [CommonModule,
        LpxSideMenuLayoutModule,
        RouterModule,
        LpxNavbarModule,
        CoreModule,
        LeptonXAbpCoreModule,
        PanelsModule,
        UserProfileModule,
        LanguageSelectionModule,
        LpxTranslateModule,
        LpxFooterModule] });
SideMenuLayoutModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: SideMenuLayoutModule, imports: [CommonModule,
        LpxSideMenuLayoutModule,
        RouterModule,
        LpxNavbarModule,
        CoreModule,
        LeptonXAbpCoreModule,
        PanelsModule,
        UserProfileModule,
        LanguageSelectionModule,
        LpxTranslateModule,
        LpxFooterModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: SideMenuLayoutModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        SideMenuApplicationLayoutComponent,
                        NavItemsComponent,
                        LanguageSelectionComponent,
                        UserProfileComponent,
                        NavigateToLoginComponent,
                    ],
                    imports: [
                        CommonModule,
                        LpxSideMenuLayoutModule,
                        RouterModule,
                        LpxNavbarModule,
                        CoreModule,
                        LeptonXAbpCoreModule,
                        PanelsModule,
                        UserProfileModule,
                        LanguageSelectionModule,
                        LpxTranslateModule,
                        LpxFooterModule,
                    ],
                }]
        }] });

/*
 * Public API Surface of theme-lepton-x/layouts
 */

/**
 * Generated bundle index. Do not edit.
 */

export { LPX_LAYOUT_PROVIDER, LPX_NAVBAR_ITEMS_PROVIDER, LanguageSelectionComponent, NAV_ITEM_PROVIDER, NavItemsComponent, NavigateToLoginComponent, SideMenuApplicationLayoutComponent, SideMenuLayoutModule, UserProfileComponent, addNavItems, initLayouts };
