// @abp/ng.schematics/utils/angular-schematic/generate-lib.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.addLibToWorkspaceFile = exports.updateTsConfig = void 0;
const angular_1 = require("../angular");
function updateTsConfig(packageName, ...paths) {
    return (host) => {
        if (!host.exists('tsconfig.json')) {
            return host;
        }
        const file = new angular_1.JSONFile(host, 'tsconfig.json');
        const jsonPath = ['compilerOptions', 'paths', packageName];
        const value = file.get(jsonPath);
        file.modify(jsonPath, Array.isArray(value) ? [...value, ...paths] : paths);
    };
}
exports.updateTsConfig = updateTsConfig;
function addLibToWorkspaceFile(projectRoot, projectName) {
    return (0, angular_1.updateWorkspace)(workspace => {
        workspace.projects.add({
            name: projectName,
            root: projectRoot,
            sourceRoot: `${projectRoot}/src`,
            projectType: angular_1.ProjectType.Library,
            prefix: 'lib',
            targets: {
                build: {
                    builder: angular_1.Builders.NgPackagr,
                    defaultConfiguration: 'production',
                    options: {
                        project: `${projectRoot}/ng-package.json`,
                    },
                    configurations: {
                        production: {
                            tsConfig: `${projectRoot}/tsconfig.lib.prod.json`,
                        },
                        development: {
                            tsConfig: `${projectRoot}/tsconfig.lib.json`,
                        },
                    },
                },
                test: {
                    builder: angular_1.Builders.Karma,
                    options: {
                        main: `${projectRoot}/src/test.ts`,
                        tsConfig: `${projectRoot}/tsconfig.spec.json`,
                        karmaConfig: `${projectRoot}/karma.conf.js`,
                    },
                },
            },
        });
    });
}
exports.addLibToWorkspaceFile = addLibToWorkspaceFile;
