// @volo/ngx-lepton-x.core/esm2020/lib/directives/click-outside/click-outside.directive.mjs
import { Directive, Output, EventEmitter, HostListener, ElementRef, Input } from '@angular/core';
import * as i0 from "@angular/core";
export class ClickOutsideDirective {
    constructor(elementRef) {
        this.elementRef = elementRef;
        this.lpxClickOutside = new EventEmitter();
        this.exceptedRefs = [];
    }
    onDocumentClick(event) {
        if (!(this.elementRef.nativeElement.contains(event.target) ||
            this.exceptedRefs.some(ref => ref.nativeElement.contains(event.target)))) {
            this.lpxClickOutside.emit();
        }
    }
}
ClickOutsideDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: ClickOutsideDirective, deps: [{ token: i0.ElementRef }], target: i0.ɵɵFactoryTarget.Directive });
ClickOutsideDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "14.2.0", type: ClickOutsideDirective, selector: "[lpxClickOutside]", inputs: { exceptedRefs: "exceptedRefs" }, outputs: { lpxClickOutside: "lpxClickOutside" }, host: { listeners: { "document:click": "onDocumentClick($event)" } }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: ClickOutsideDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[lpxClickOutside]',
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }]; }, propDecorators: { lpxClickOutside: [{
                type: Output
            }], exceptedRefs: [{
                type: Input
            }], onDocumentClick: [{
                type: HostListener,
                args: ['document:click', ['$event']]
            }] } });
