// @volo/ngx-lepton-x.lite/esm2020/layouts/lib/components/mn-language-selection/mn-language-selection.component.mjs
import { Component, Inject, ViewEncapsulation } from '@angular/core';
import { LanguageService } from '@volo/ngx-lepton-x.core';
import { OBSERVE, OBSERVE_PROVIDER } from 'ng-observe';
import * as i0 from "@angular/core";
import * as i1 from "@volo/ngx-lepton-x.core";
import * as i2 from "@angular/common";
export class MnLanguageSelectionComponent {
    constructor(service, observe) {
        this.service = service;
        this.observe = observe;
        this.availableLanguages = this.observe(this.service.languages$);
        this.selectedLanguage = this.observe(this.service.selectedLanguage$);
        this.menuHidden = true;
    }
    get langs() {
        return this.availableLanguages.value;
    }
    toggleMenu() {
        this.menuHidden = !this.menuHidden;
    }
    onLanguageSelection(lang) {
        this.service.setSelectedLanguage(lang);
    }
}
MnLanguageSelectionComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: MnLanguageSelectionComponent, deps: [{ token: i1.LanguageService }, { token: OBSERVE }], target: i0.ɵɵFactoryTarget.Component });
MnLanguageSelectionComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "14.2.0", type: MnLanguageSelectionComponent, selector: "lpx-mn-language-selection", providers: [OBSERVE_PROVIDER], ngImport: i0, template: "<div class=\"lpx-language-selection btn-toggle\" role=\"button\" (click)=\"toggleMenu()\">\r\n  <lpx-icon iconClass=\"bi bi-globe\"></lpx-icon>\r\n  {{ selectedLanguage.value?.displayName }}\r\n</div>\r\n\r\n<div class=\"action-group-menu\" [class.d-none]=\"menuHidden\">\r\n  <a class=\"action\" role=\"button\" (click)=\"onLanguageSelection(lang)\" *ngFor=\"let lang of langs\">\r\n    {{ lang.displayName }}\r\n  </a>\r\n</div>\r\n", dependencies: [{ kind: "directive", type: i2.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }, { kind: "component", type: i1.IconComponent, selector: "lpx-icon", inputs: ["iconClass"] }], encapsulation: i0.ViewEncapsulation.None });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: MnLanguageSelectionComponent, decorators: [{
            type: Component,
            args: [{ selector: 'lpx-mn-language-selection', encapsulation: ViewEncapsulation.None, providers: [OBSERVE_PROVIDER], template: "<div class=\"lpx-language-selection btn-toggle\" role=\"button\" (click)=\"toggleMenu()\">\r\n  <lpx-icon iconClass=\"bi bi-globe\"></lpx-icon>\r\n  {{ selectedLanguage.value?.displayName }}\r\n</div>\r\n\r\n<div class=\"action-group-menu\" [class.d-none]=\"menuHidden\">\r\n  <a class=\"action\" role=\"button\" (click)=\"onLanguageSelection(lang)\" *ngFor=\"let lang of langs\">\r\n    {{ lang.displayName }}\r\n  </a>\r\n</div>\r\n" }]
        }], ctorParameters: function () { return [{ type: i1.LanguageService }, { type: undefined, decorators: [{
                    type: Inject,
                    args: [OBSERVE]
                }] }]; } });
