// @volo/ngx-lepton-x.lite/esm2020/layouts/lib/components/mn-language-selection/mn-language-selection.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { LpxIconModule } from '@volo/ngx-lepton-x.core';
import { MnLanguageSelectionComponent } from './mn-language-selection.component';
import * as i0 from "@angular/core";
export class MnLanguageSelectionModule {
}
MnLanguageSelectionModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: MnLanguageSelectionModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MnLanguageSelectionModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.2.0", ngImport: i0, type: MnLanguageSelectionModule, declarations: [MnLanguageSelectionComponent], imports: [CommonModule, LpxIconModule], exports: [MnLanguageSelectionComponent] });
MnLanguageSelectionModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: MnLanguageSelectionModule, imports: [CommonModule, LpxIconModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: MnLanguageSelectionModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [MnLanguageSelectionComponent],
                    imports: [CommonModule, LpxIconModule],
                    exports: [MnLanguageSelectionComponent],
                }]
        }] });
