// @volo/ngx-lepton-x.core/esm2020/lib/components/navbar/navbar.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { RouterModule } from '@angular/router';
import { LpxBrandLogoModule } from '../brand-logo/brand-logo.module';
import { LpxIconModule } from '../icon/icon.module';
import { NavbarComponent } from './navbar.component';
import { SubNavbarComponent } from './sub-navbar/sub-navbar.component';
import { NavbarRoutesComponent } from './navbar-routes/navbar-routes.component';
import { NavbarRoutesDirective } from './navbar-routes/navbar-routes.directive';
import { CONTENT_AFTER_ROUTES, CONTENT_BEFORE_ROUTES, LPX_MENU_ITEMS, } from './navbar.token';
import { ToObservableModule } from '../../pipes/to-observable/to-observable.module';
import { LpxTranslateModule } from '../../pipes';
import * as i0 from "@angular/core";
const exportedDeclarations = [
    NavbarComponent,
    SubNavbarComponent,
    NavbarRoutesComponent,
    NavbarRoutesDirective,
];
export class LpxNavbarModule {
    static forRoot(options = {}) {
        return {
            ngModule: LpxNavbarModule,
            providers: [
                {
                    provide: LPX_MENU_ITEMS,
                    useValue: options?.menuItems || [],
                },
                {
                    provide: CONTENT_AFTER_ROUTES,
                    useValue: options?.contentAfterRoutes || [],
                    multi: true,
                },
                {
                    provide: CONTENT_BEFORE_ROUTES,
                    useValue: options?.contentBeforeRoutes || [],
                    multi: true,
                },
            ],
        };
    }
    static forChild(options = {}) {
        return {
            ngModule: LpxNavbarModule,
            providers: [
                {
                    provide: CONTENT_AFTER_ROUTES,
                    useValue: options?.contentAfterRoutes || [],
                    multi: true,
                },
                {
                    provide: CONTENT_BEFORE_ROUTES,
                    useValue: options?.contentBeforeRoutes || [],
                    multi: true,
                },
            ],
        };
    }
}
LpxNavbarModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: LpxNavbarModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
LpxNavbarModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.2.0", ngImport: i0, type: LpxNavbarModule, declarations: [NavbarComponent,
        SubNavbarComponent,
        NavbarRoutesComponent,
        NavbarRoutesDirective], imports: [CommonModule,
        FormsModule,
        RouterModule,
        LpxBrandLogoModule,
        LpxIconModule,
        ToObservableModule,
        LpxTranslateModule], exports: [NavbarComponent,
        SubNavbarComponent,
        NavbarRoutesComponent,
        NavbarRoutesDirective] });
LpxNavbarModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: LpxNavbarModule, imports: [CommonModule,
        FormsModule,
        RouterModule,
        LpxBrandLogoModule,
        LpxIconModule,
        ToObservableModule,
        LpxTranslateModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: LpxNavbarModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [...exportedDeclarations],
                    imports: [
                        CommonModule,
                        FormsModule,
                        RouterModule,
                        LpxBrandLogoModule,
                        LpxIconModule,
                        ToObservableModule,
                        LpxTranslateModule
                    ],
                    exports: [...exportedDeclarations],
                }]
        }] });
