// @volo/ngx-lepton-x.core/esm2020/lib/components/navbar/navbar-routes/navbar-routes.component.mjs
import { Component, EventEmitter, Input, Output, ViewEncapsulation, } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
import * as i2 from "../sub-navbar/sub-navbar.component";
export class NavbarRoutesComponent {
    constructor() {
        this.routeClick = new EventEmitter();
    }
    onSubnavbarExpand(menuItem, menuItems) {
        if (menuItem.expanded) {
            menuItems
                ?.filter((item) => item !== menuItem)
                .forEach((item) => (item.expanded = false));
        }
    }
    onRouteClick(menuItem, menuItems) {
        const expandedItems = menuItems?.filter((item) => (item.expanded || item.selected) &&
            (!item.children || item.children?.includes(menuItem)));
        if (expandedItems) {
            expandedItems
                .filter((item) => item !== menuItem)
                .reduce((acc, item) => {
                item.selected = false;
                return [...acc, ...(item.children || [])];
            }, [])
                ?.filter((item) => item !== menuItem)
                .forEach((item) => {
                item.selected = false;
                item.expanded = false;
            });
        }
        this.routeClick.emit(menuItem);
    }
}
NavbarRoutesComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: NavbarRoutesComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
NavbarRoutesComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "14.2.0", type: NavbarRoutesComponent, selector: "lpx-navbar-routes", inputs: { navbarItems: "navbarItems", routerItem: "routerItem" }, outputs: { routeClick: "routeClick" }, ngImport: i0, template: "<ul class=\"lpx-nav-menu\">\r\n  <li class=\"outer-menu-item\" *ngFor=\"let item of navbarItems\">\r\n    <lpx-sub-navbar\r\n      [item]=\"item\"\r\n      (expand)=\"onSubnavbarExpand($event, navbarItems)\"\r\n      (routeClick)=\"onRouteClick($event, navbarItems)\"\r\n      [routerItem]=\"routerItem\"\r\n    ></lpx-sub-navbar>\r\n  </li>\r\n</ul>\r\n", dependencies: [{ kind: "directive", type: i1.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }, { kind: "component", type: i2.SubNavbarComponent, selector: "lpx-sub-navbar", inputs: ["item", "routerItem"], outputs: ["routeClick", "expand"] }], encapsulation: i0.ViewEncapsulation.None });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: NavbarRoutesComponent, decorators: [{
            type: Component,
            args: [{ selector: 'lpx-navbar-routes', encapsulation: ViewEncapsulation.None, template: "<ul class=\"lpx-nav-menu\">\r\n  <li class=\"outer-menu-item\" *ngFor=\"let item of navbarItems\">\r\n    <lpx-sub-navbar\r\n      [item]=\"item\"\r\n      (expand)=\"onSubnavbarExpand($event, navbarItems)\"\r\n      (routeClick)=\"onRouteClick($event, navbarItems)\"\r\n      [routerItem]=\"routerItem\"\r\n    ></lpx-sub-navbar>\r\n  </li>\r\n</ul>\r\n" }]
        }], propDecorators: { navbarItems: [{
                type: Input
            }], routerItem: [{
                type: Input
            }], routeClick: [{
                type: Output
            }] } });
