// @volo/ngx-lepton-x.core/esm2020/lib/services/user-profile/user-profile.service.mjs
import { Injectable } from '@angular/core';
import { DataStore } from '../../utils/data-store';
import * as i0 from "@angular/core";
export class UserProfileService {
    constructor() {
        this.store = new DataStore({});
        this.user$ = this.store.sliceState((state) => state);
    }
    setUser(user) {
        this.store.set(user);
    }
    patchUser(user) {
        this.store.patch(user);
    }
}
UserProfileService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: UserProfileService, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
UserProfileService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: UserProfileService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: UserProfileService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }] });
