// @volo/ngx-lepton-x.lite/esm2020/layouts/lib/components/mn-user-profile/mn-user-profile.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { LpxAvatarModule, LpxTranslateModule } from '@volo/ngx-lepton-x.core';
import { RouterModule } from '@angular/router';
import { MnUserProfileComponent } from './mn-user-profile.component';
import * as i0 from "@angular/core";
export class MnUserProfileModule {
}
MnUserProfileModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: MnUserProfileModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MnUserProfileModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.2.0", ngImport: i0, type: MnUserProfileModule, declarations: [MnUserProfileComponent], imports: [CommonModule, LpxAvatarModule, RouterModule, LpxTranslateModule], exports: [MnUserProfileComponent] });
MnUserProfileModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: MnUserProfileModule, imports: [CommonModule, LpxAvatarModule, RouterModule, LpxTranslateModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: MnUserProfileModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [MnUserProfileComponent],
                    imports: [CommonModule, LpxAvatarModule, RouterModule, LpxTranslateModule],
                    exports: [MnUserProfileComponent],
                }]
        }] });
