// @volo/abp.ng.lepton-x.core/fesm2020/volo-abp.ng.lepton-x.core.mjs
import * as i0 from '@angular/core';
import { InjectionToken, Injectable, Inject, APP_INITIALIZER, Directive, Input, Component, NgModule } from '@angular/core';
import * as i2 from '@angular/common';
import { DOCUMENT, CommonModule } from '@angular/common';
import * as i1 from '@abp/ng.theme.shared';
import { ThemeSharedModule } from '@abp/ng.theme.shared';
import * as i1$1 from '@abp/ng.core';
import { CoreModule } from '@abp/ng.core';
import * as i2$1 from '@volo/ngx-lepton-x.core';
import { map } from 'rxjs/operators';

const MOBILE_NAVBAR_ITEMS_FILTER_TOKEN = new InjectionToken('MOBILE_NAVBAR_ITEMS_FILTER_TOKEN');

const MOBILE_NAVBAR_ITEMS_FILTER_PROVIDER = {
    provide: MOBILE_NAVBAR_ITEMS_FILTER_TOKEN,
    deps: [],
    useFactory: mobileMenuItemFilterFnFactory
};
function mobileMenuItemFilterFnFactory() {
    return (route, index) => {
        return index === 0 || index === 1;
    };
}

class DocumentDirService {
    constructor(documentDirHandler, document) {
        this.documentDirHandler = documentDirHandler;
        this.document = document;
    }
    listenDir() {
        this.documentDirHandler.dir$.subscribe((dir) => {
            this.document.documentElement.dir = dir;
        });
    }
}
DocumentDirService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: DocumentDirService, deps: [{ token: i1.DocumentDirHandlerService }, { token: DOCUMENT }], target: i0.ɵɵFactoryTarget.Injectable });
DocumentDirService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: DocumentDirService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: DocumentDirService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.DocumentDirHandlerService }, { type: Document, decorators: [{
                    type: Inject,
                    args: [DOCUMENT]
                }] }]; } });

const DOCUMENT_DIR_PROVIDER = {
    provide: APP_INITIALIZER,
    multi: true,
    deps: [DocumentDirService],
    useFactory: listenDirectionChangeFromAbp
};
function listenDirectionChangeFromAbp(documentDirService) {
    return () => {
        documentDirService.listenDir();
    };
}

class IfReplaceableTemplateExistsDirective {
    constructor(view, template, replaceableComponentsService) {
        this.view = view;
        this.template = template;
        this.replaceableComponentsService = replaceableComponentsService;
    }
    ngAfterViewInit() {
        const replaceableComponentInstance = this.replaceableComponentsService.get(this.abpIfReplaceableTemplateExists);
        const isReplaceableComponentInstanceNotExits = !replaceableComponentInstance;
        if (isReplaceableComponentInstanceNotExits) {
            return;
        }
        this.view.createEmbeddedView(this.template);
    }
}
IfReplaceableTemplateExistsDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: IfReplaceableTemplateExistsDirective, deps: [{ token: i0.ViewContainerRef }, { token: i0.TemplateRef }, { token: i1$1.ReplaceableComponentsService }], target: i0.ɵɵFactoryTarget.Directive });
IfReplaceableTemplateExistsDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "14.2.0", type: IfReplaceableTemplateExistsDirective, selector: "[abpIfReplaceableTemplateExists]", inputs: { abpIfReplaceableTemplateExists: "abpIfReplaceableTemplateExists" }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: IfReplaceableTemplateExistsDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[abpIfReplaceableTemplateExists]',
                }]
        }], ctorParameters: function () { return [{ type: i0.ViewContainerRef }, { type: i0.TemplateRef }, { type: i1$1.ReplaceableComponentsService }]; }, propDecorators: { abpIfReplaceableTemplateExists: [{
                type: Input
            }] } });

class PageAlertContainerComponent {
    constructor(service) {
        this.service = service;
        this.replaceableTemplateKey = {
            componentKey: 'Theme.PageAlertContainerComponent',
        };
    }
}
PageAlertContainerComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: PageAlertContainerComponent, deps: [{ token: i1.PageAlertService }], target: i0.ɵɵFactoryTarget.Component });
PageAlertContainerComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "14.2.0", type: PageAlertContainerComponent, selector: "abp-page-alert-container", ngImport: i0, template: "<ng-container *abpReplaceableTemplate=\"replaceableTemplateKey\">\r\n  <ng-container *ngFor=\"let alert of service.alerts$ | async; let i = index\">\r\n    <div\r\n      class=\"alert alert-{{ alert.type }} fade show\"\r\n      [ngClass]=\"{ 'alert-dismissible fade show': alert.dismissible }\"\r\n      role=\"alert\"\r\n    >\r\n      <h4 class=\"alert-heading\" *ngIf=\"alert.title\">\r\n        {{ alert.title | abpLocalization: alert.titleLocalizationParams }}\r\n      </h4>\r\n      {{ alert.message | abpLocalization: alert.messageLocalizationParams }}\r\n      <button\r\n        *ngIf=\"alert.dismissible\"\r\n        type=\"button\"\r\n        class=\"btn-close\"\r\n        data-dismiss=\"alert\"\r\n        aria-label=\"Close\"\r\n        (click)=\"service.remove(i)\"\r\n      ></button>\r\n    </div>\r\n  </ng-container>\r\n</ng-container>\r\n", dependencies: [{ kind: "directive", type: i2.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }, { kind: "directive", type: i2.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }, { kind: "directive", type: i2.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { kind: "directive", type: i1$1.ReplaceableTemplateDirective, selector: "[abpReplaceableTemplate]", inputs: ["abpReplaceableTemplate"] }, { kind: "pipe", type: i2.AsyncPipe, name: "async" }, { kind: "pipe", type: i1$1.LocalizationPipe, name: "abpLocalization" }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: PageAlertContainerComponent, decorators: [{
            type: Component,
            args: [{ selector: 'abp-page-alert-container', template: "<ng-container *abpReplaceableTemplate=\"replaceableTemplateKey\">\r\n  <ng-container *ngFor=\"let alert of service.alerts$ | async; let i = index\">\r\n    <div\r\n      class=\"alert alert-{{ alert.type }} fade show\"\r\n      [ngClass]=\"{ 'alert-dismissible fade show': alert.dismissible }\"\r\n      role=\"alert\"\r\n    >\r\n      <h4 class=\"alert-heading\" *ngIf=\"alert.title\">\r\n        {{ alert.title | abpLocalization: alert.titleLocalizationParams }}\r\n      </h4>\r\n      {{ alert.message | abpLocalization: alert.messageLocalizationParams }}\r\n      <button\r\n        *ngIf=\"alert.dismissible\"\r\n        type=\"button\"\r\n        class=\"btn-close\"\r\n        data-dismiss=\"alert\"\r\n        aria-label=\"Close\"\r\n        (click)=\"service.remove(i)\"\r\n      ></button>\r\n    </div>\r\n  </ng-container>\r\n</ng-container>\r\n" }]
        }], ctorParameters: function () { return [{ type: i1.PageAlertService }]; } });

class PageAlertContainerModule {
}
PageAlertContainerModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: PageAlertContainerModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
PageAlertContainerModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.2.0", ngImport: i0, type: PageAlertContainerModule, declarations: [PageAlertContainerComponent], imports: [ThemeSharedModule, CoreModule], exports: [PageAlertContainerComponent] });
PageAlertContainerModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: PageAlertContainerModule, imports: [ThemeSharedModule, CoreModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: PageAlertContainerModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [PageAlertContainerComponent],
                    imports: [ThemeSharedModule, CoreModule],
                    exports: [PageAlertContainerComponent],
                }]
        }] });

class LeptonXAbpCoreModule {
    static forRoot() {
        return {
            ngModule: LeptonXAbpCoreModule,
            providers: [DOCUMENT_DIR_PROVIDER, MOBILE_NAVBAR_ITEMS_FILTER_PROVIDER],
        };
    }
}
LeptonXAbpCoreModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: LeptonXAbpCoreModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
LeptonXAbpCoreModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.2.0", ngImport: i0, type: LeptonXAbpCoreModule, declarations: [IfReplaceableTemplateExistsDirective], imports: [CommonModule, PageAlertContainerModule], exports: [IfReplaceableTemplateExistsDirective, PageAlertContainerModule] });
LeptonXAbpCoreModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: LeptonXAbpCoreModule, imports: [CommonModule, PageAlertContainerModule, PageAlertContainerModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: LeptonXAbpCoreModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, PageAlertContainerModule],
                    declarations: [IfReplaceableTemplateExistsDirective],
                    exports: [IfReplaceableTemplateExistsDirective, PageAlertContainerModule],
                }]
        }] });

class AbpNavbarService {
    constructor(routes, navbarService, localizationService, mobileMenuItemFilterFn) {
        this.routes = routes;
        this.navbarService = navbarService;
        this.localizationService = localizationService;
        this.mobileMenuItemFilterFn = mobileMenuItemFilterFn;
        this.mapRouteToNavItem = (route, index) => {
            return {
                text: this.localizationService.instant(route.name),
                link: route.children && route.children.length ? undefined : route.path,
                icon: route.iconClass,
                children: this.getRouteChildrenAsNavItems(route.children || []),
                showOnMobileNavbar: this.mobileMenuItemFilterFn(route, index),
            };
        };
    }
    initRoutes() {
        this.routes.visible$
            .pipe(map((routes) => routes.filter((route) => route.path || route.children.length)))
            .subscribe((routes) => {
            this.navbarService.setNavbarItems(...routes.map(this.mapRouteToNavItem));
        });
    }
    getRouteChildrenAsNavItems(children) {
        return children.map(this.mapRouteToNavItem);
    }
}
AbpNavbarService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: AbpNavbarService, deps: [{ token: i1$1.RoutesService }, { token: i2$1.NavbarService }, { token: i1$1.LocalizationService }, { token: MOBILE_NAVBAR_ITEMS_FILTER_TOKEN }], target: i0.ɵɵFactoryTarget.Injectable });
AbpNavbarService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: AbpNavbarService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: AbpNavbarService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1$1.RoutesService }, { type: i2$1.NavbarService }, { type: i1$1.LocalizationService }, { type: undefined, decorators: [{
                    type: Inject,
                    args: [MOBILE_NAVBAR_ITEMS_FILTER_TOKEN]
                }] }]; } });

/**
 * Generated bundle index. Do not edit.
 */

export { AbpNavbarService, IfReplaceableTemplateExistsDirective, LeptonXAbpCoreModule, MOBILE_NAVBAR_ITEMS_FILTER_TOKEN, PageAlertContainerComponent, PageAlertContainerModule };
