// @abp/ng.theme.lepton-x/esm2020/account/account-layout/account-layout.module.mjs
import { APP_INITIALIZER, Injector, NgModule, } from '@angular/core';
import { AccountLayoutComponent } from './account-layout.component';
import { TenantBoxComponent } from './components/tenant-box/tenant-box.component';
import { AuthWrapperComponent } from './components/auth-layout/auth-wrapper.component';
import { CoreModule, ReplaceableComponentsService } from '@abp/ng.core';
import { LpxSideMenuLayoutModule } from '@volo/ngx-lepton-x.lite/layouts';
import { LanguageSelectionModule } from '@volo/ngx-lepton-x.lite';
import { RouterModule } from '@angular/router';
import { ThemeSharedModule } from '@abp/ng.theme.shared';
import { PageAlertContainerModule } from '@volo/abp.ng.lepton-x.core';
import * as i0 from "@angular/core";
export class AccountLayoutModule {
    static forRoot() {
        return {
            ngModule: AccountLayoutModule,
            providers: [
                {
                    provide: APP_INITIALIZER,
                    useFactory: initAccountLayout,
                    deps: [Injector],
                    multi: true,
                },
            ],
        };
    }
}
AccountLayoutModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: AccountLayoutModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
AccountLayoutModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.2.0", ngImport: i0, type: AccountLayoutModule, declarations: [AccountLayoutComponent,
        TenantBoxComponent,
        AuthWrapperComponent], imports: [RouterModule,
        CoreModule,
        LpxSideMenuLayoutModule,
        ThemeSharedModule,
        PageAlertContainerModule,
        LanguageSelectionModule] });
AccountLayoutModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: AccountLayoutModule, imports: [RouterModule,
        CoreModule,
        LpxSideMenuLayoutModule,
        ThemeSharedModule,
        PageAlertContainerModule,
        LanguageSelectionModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: AccountLayoutModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        AccountLayoutComponent,
                        TenantBoxComponent,
                        AuthWrapperComponent,
                    ],
                    imports: [
                        RouterModule,
                        CoreModule,
                        LpxSideMenuLayoutModule,
                        ThemeSharedModule,
                        PageAlertContainerModule,
                        LanguageSelectionModule,
                    ],
                }]
        }] });
export function initAccountLayout(injector) {
    function init() {
        const replaceableComponents = injector.get(ReplaceableComponentsService);
        replaceableComponents.add({
            key: "Theme.AccountLayoutComponent" /* eThemeLeptonXComponents.AccountLayout */,
            component: AccountLayoutComponent,
        });
    }
    return init;
}
