// @volo/ngx-lepton-x.core/esm2020/lib/components/brand-logo/brand-logo.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { BrandLogoComponent } from './brand-logo.component';
import * as i0 from "@angular/core";
export class LpxBrandLogoModule {
}
LpxBrandLogoModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: LpxBrandLogoModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
LpxBrandLogoModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.2.0", ngImport: i0, type: LpxBrandLogoModule, declarations: [BrandLogoComponent], imports: [CommonModule], exports: [BrandLogoComponent] });
LpxBrandLogoModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: LpxBrandLogoModule, imports: [CommonModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: LpxBrandLogoModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [BrandLogoComponent],
                    imports: [
                        CommonModule
                    ],
                    exports: [BrandLogoComponent]
                }]
        }] });
