// @volo/abp.ng.lepton-x.core/esm2020/lib/components/page-alert-container/page-alert-container.module.mjs
import { NgModule } from '@angular/core';
import { ThemeSharedModule } from '@abp/ng.theme.shared';
import { CoreModule } from '@abp/ng.core';
import { PageAlertContainerComponent } from './page-alert-container.component';
import * as i0 from "@angular/core";
export class PageAlertContainerModule {
}
PageAlertContainerModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: PageAlertContainerModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
PageAlertContainerModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.2.0", ngImport: i0, type: PageAlertContainerModule, declarations: [PageAlertContainerComponent], imports: [ThemeSharedModule, CoreModule], exports: [PageAlertContainerComponent] });
PageAlertContainerModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: PageAlertContainerModule, imports: [ThemeSharedModule, CoreModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: PageAlertContainerModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [PageAlertContainerComponent],
                    imports: [ThemeSharedModule, CoreModule],
                    exports: [PageAlertContainerComponent],
                }]
        }] });
