// @abp/ng.feature-management/esm2020/lib/feature-management.module.mjs
import { CoreModule } from '@abp/ng.core';
import { ThemeSharedModule } from '@abp/ng.theme.shared';
import { NgModule } from '@angular/core';
import { NgbNavModule } from '@ng-bootstrap/ng-bootstrap';
import { FeatureManagementComponent } from './components/feature-management/feature-management.component';
import { FreeTextInputDirective } from './directives/free-text-input.directive';
import { FEATURE_MANAGEMENT_SETTINGS_PROVIDERS } from './providers';
import { FeatureManagementTabComponent } from './components';
import * as i0 from "@angular/core";
const exported = [
    FeatureManagementComponent,
    FreeTextInputDirective,
    FeatureManagementTabComponent,
];
export class FeatureManagementModule {
    static forRoot() {
        return {
            ngModule: FeatureManagementModule,
            providers: [FEATURE_MANAGEMENT_SETTINGS_PROVIDERS],
        };
    }
}
FeatureManagementModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.3", ngImport: i0, type: FeatureManagementModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
FeatureManagementModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.0.3", ngImport: i0, type: FeatureManagementModule, declarations: [FeatureManagementComponent,
        FreeTextInputDirective,
        FeatureManagementTabComponent], imports: [CoreModule, ThemeSharedModule, NgbNavModule], exports: [FeatureManagementComponent,
        FreeTextInputDirective,
        FeatureManagementTabComponent] });
FeatureManagementModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.0.3", ngImport: i0, type: FeatureManagementModule, imports: [CoreModule, ThemeSharedModule, NgbNavModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.3", ngImport: i0, type: FeatureManagementModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [...exported],
                    imports: [CoreModule, ThemeSharedModule, NgbNavModule],
                    exports: [...exported],
                }]
        }] });
