// @volo/ngx-lepton-x.core/esm2020/lib/lepton-x-core.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { LpxLanguageModule } from './services/language/language.module';
import { LpxIconModule } from './components/icon/icon.module';
import { LpxNavbarModule } from './components/navbar/navbar.module';
import { LpxBreadcrumbModule } from './components/breadcrumb/breadcrumb.module';
import { LPX_TRANSLATE_PROVIDERS } from './services/translate/providers';
import { LPX_STYLE_PROVIDERS } from './providers/styles.provider';
import { createWindowProvider } from './providers/window.provider';
import { createResponsiveProvider } from './providers/responsive.provider';
import { createDirectionProvider } from './providers/direction.provider';
import * as i0 from "@angular/core";
export class LpxCoreModule {
    static forRoot(options) {
        return {
            ngModule: LpxCoreModule,
            providers: [
                createResponsiveProvider(options?.responsiveSettings),
                createWindowProvider(options?.window),
                LpxIconModule.forRoot(options?.iconSettings).providers,
                LpxLanguageModule.forRoot(options?.languageSettings).providers,
                LpxNavbarModule.forRoot(options?.navbarSettings).providers,
                LpxBreadcrumbModule.forRoot().providers,
                LPX_TRANSLATE_PROVIDERS,
                ...LPX_STYLE_PROVIDERS,
                createDirectionProvider(options?.listenDirectionChanges || true),
            ],
        };
    }
}
LpxCoreModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: LpxCoreModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
LpxCoreModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.2.0", ngImport: i0, type: LpxCoreModule, imports: [CommonModule] });
LpxCoreModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: LpxCoreModule, imports: [CommonModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: LpxCoreModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                }]
        }] });
