// @volo/ngx-lepton-x.lite/esm2020/lib/components/user-profile/user-profile.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';
import { NgbDropdownModule } from '@ng-bootstrap/ng-bootstrap';
import { LpxAvatarModule, LpxTranslateModule } from '@volo/ngx-lepton-x.core';
import { UserProfileComponent } from './user-profile.component';
import * as i0 from "@angular/core";
export class UserProfileModule {
}
UserProfileModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: UserProfileModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
UserProfileModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.2.0", ngImport: i0, type: UserProfileModule, declarations: [UserProfileComponent], imports: [CommonModule,
        LpxAvatarModule,
        NgbDropdownModule,
        RouterModule,
        LpxTranslateModule], exports: [UserProfileComponent] });
UserProfileModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: UserProfileModule, imports: [CommonModule,
        LpxAvatarModule,
        NgbDropdownModule,
        RouterModule,
        LpxTranslateModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: UserProfileModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [UserProfileComponent],
                    imports: [
                        CommonModule,
                        LpxAvatarModule,
                        NgbDropdownModule,
                        RouterModule,
                        LpxTranslateModule,
                    ],
                    exports: [UserProfileComponent],
                }]
        }] });
