// @abp/ng.theme.lepton-x/esm2020/lib/services/user-profile.service.mjs
import { Inject, Injectable } from '@angular/core';
import { combineLatest, of } from 'rxjs';
import { AuthService, ConfigStateService, NAVIGATE_TO_MANAGE_PROFILE, } from '@abp/ng.core';
import { UserProfileService } from '@volo/ngx-lepton-x.core';
import { filter } from 'rxjs/operators';
import { OAuthService } from 'angular-oauth2-oidc';
import { UserMenuService } from '@abp/ng.theme.shared';
import * as i0 from "@angular/core";
import * as i1 from "@abp/ng.core";
import * as i2 from "@volo/ngx-lepton-x.core";
import * as i3 from "angular-oauth2-oidc";
import * as i4 from "@abp/ng.theme.shared";
export class AbpUserProfileService {
    constructor(configState, userProfileService, authService, navigateToManageProfile, oAuthService, userMenuService) {
        this.configState = configState;
        this.userProfileService = userProfileService;
        this.authService = authService;
        this.navigateToManageProfile = navigateToManageProfile;
        this.oAuthService = oAuthService;
        this.userMenuService = userMenuService;
        this.currentUser$ = this.configState.getOne$('currentUser');
    }
    subscribeUser() {
        combineLatest([
            this.currentUser$.pipe(filter(Boolean)),
            this.userMenuService.items$,
        ]).subscribe(([user, userMenuItems]) => {
            const userActionGroups = userMenuItems.reduce((acc, curr) => {
                const menuItem = {
                    icon: curr.textTemplate?.icon,
                    text: curr.textTemplate?.text,
                    component: curr?.component,
                    action: () => {
                        curr.action();
                        return of(true);
                    },
                };
                acc[0].push(menuItem);
                return acc;
            }, [[]]);
            this.userProfileService.setUser({
                fullName: user.name || user.userName || '',
                email: user.email || '',
                userName: user.userName || '',
                avatar: {
                    type: 'icon',
                    source: 'bi bi-person-circle',
                },
                userActionGroups,
            });
        });
    }
}
AbpUserProfileService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: AbpUserProfileService, deps: [{ token: i1.ConfigStateService }, { token: i2.UserProfileService }, { token: i1.AuthService }, { token: NAVIGATE_TO_MANAGE_PROFILE }, { token: i3.OAuthService }, { token: i4.UserMenuService }], target: i0.ɵɵFactoryTarget.Injectable });
AbpUserProfileService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: AbpUserProfileService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: AbpUserProfileService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.ConfigStateService }, { type: i2.UserProfileService }, { type: i1.AuthService }, { type: undefined, decorators: [{
                    type: Inject,
                    args: [NAVIGATE_TO_MANAGE_PROFILE]
                }] }, { type: i3.OAuthService }, { type: i4.UserMenuService }]; } });
