// @volo/ngx-lepton-x.core/esm2020/lib/services/body/body.service.mjs
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
export class BodyService {
    constructor() {
        this.body = document.querySelector('body');
        this.classes = {
            overflowYHidden: 'overflow-y-hidden',
        };
    }
    disableScrollY() {
        this.body?.classList.add(this.classes.overflowYHidden);
    }
    enableScrollY() {
        this.body?.classList.remove(this.classes.overflowYHidden);
    }
}
BodyService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: BodyService, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
BodyService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: BodyService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: BodyService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }] });
