// @volo/ngx-lepton-x.core/esm2020/lib/directives/responsive/responsive.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ResponsiveDirective } from './responsive.directive';
import * as i0 from "@angular/core";
export class LpxResponsiveModule {
}
LpxResponsiveModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: LpxResponsiveModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
LpxResponsiveModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.2.0", ngImport: i0, type: LpxResponsiveModule, declarations: [ResponsiveDirective], imports: [CommonModule], exports: [ResponsiveDirective] });
LpxResponsiveModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: LpxResponsiveModule, imports: [CommonModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: LpxResponsiveModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [ResponsiveDirective],
                    imports: [CommonModule],
                    exports: [ResponsiveDirective],
                }]
        }] });
