// @volo/ngx-lepton-x.lite/esm2020/lib/lepton-x-lite.module.mjs
import { NgModule } from '@angular/core';
import { LpxCoreModule, } from '@volo/ngx-lepton-x.core';
import { getLpxLiteStyleProviders } from './providers';
import * as i0 from "@angular/core";
export class LpxModule {
    static forRoot(options) {
        return {
            ngModule: LpxModule,
            providers: [
                getLpxLiteStyleProviders(options?.styleFactory),
                LpxCoreModule.forRoot(options).providers,
            ],
        };
    }
}
LpxModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: LpxModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
LpxModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.2.0", ngImport: i0, type: LpxModule, imports: [LpxCoreModule] });
LpxModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: LpxModule, imports: [LpxCoreModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: LpxModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [LpxCoreModule],
                }]
        }] });
