// @abp/ng.setting-management/esm2020/config/lib/setting-management-config.module.mjs
import { ThemeSharedModule } from '@abp/ng.theme.shared';
import { NgModule } from '@angular/core';
import { CoreModule } from '@abp/ng.core';
import { NgxValidateCoreModule } from '@ngx-validate/core';
import { SETTING_MANAGEMENT_FEATURES_PROVIDERS } from './providers/features.token';
import { SETTING_MANAGEMENT_VISIBLE_PROVIDERS } from './providers/visible.provider';
import { SETTING_MANAGEMENT_ROUTE_PROVIDERS } from './providers/route.provider';
import { SETTING_MANAGEMENT_SETTING_TAB_PROVIDERS } from './providers/setting-tab.provider';
import { EmailSettingGroupComponent } from './components/email-setting-group/email-setting-group.component';
import * as i0 from "@angular/core";
export class SettingManagementConfigModule {
    static forRoot() {
        return {
            ngModule: SettingManagementConfigModule,
            providers: [
                SETTING_MANAGEMENT_ROUTE_PROVIDERS,
                SETTING_MANAGEMENT_SETTING_TAB_PROVIDERS,
                SETTING_MANAGEMENT_FEATURES_PROVIDERS,
                SETTING_MANAGEMENT_VISIBLE_PROVIDERS,
            ],
        };
    }
}
SettingManagementConfigModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.3", ngImport: i0, type: SettingManagementConfigModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
SettingManagementConfigModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.0.3", ngImport: i0, type: SettingManagementConfigModule, declarations: [EmailSettingGroupComponent], imports: [CoreModule, ThemeSharedModule, NgxValidateCoreModule], exports: [EmailSettingGroupComponent] });
SettingManagementConfigModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.0.3", ngImport: i0, type: SettingManagementConfigModule, imports: [CoreModule, ThemeSharedModule, NgxValidateCoreModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.3", ngImport: i0, type: SettingManagementConfigModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CoreModule, ThemeSharedModule, NgxValidateCoreModule],
                    declarations: [EmailSettingGroupComponent],
                    exports: [EmailSettingGroupComponent],
                }]
        }] });
