// @volo/abp.ng.lepton-x.core/esm2020/lib/lepton-x-abp-core.module.mjs
import { MOBILE_NAVBAR_ITEMS_FILTER_PROVIDER } from './providers/mobile-navbar-items-filter.provider';
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { DOCUMENT_DIR_PROVIDER } from './providers/document-dir.provider';
import { IfReplaceableTemplateExistsDirective } from './directives/if-replaceable-template-exists.directive';
import { PageAlertContainerModule } from './components/page-alert-container/page-alert-container.module';
import * as i0 from "@angular/core";
export class LeptonXAbpCoreModule {
    static forRoot() {
        return {
            ngModule: LeptonXAbpCoreModule,
            providers: [DOCUMENT_DIR_PROVIDER, MOBILE_NAVBAR_ITEMS_FILTER_PROVIDER],
        };
    }
}
LeptonXAbpCoreModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: LeptonXAbpCoreModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
LeptonXAbpCoreModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.2.0", ngImport: i0, type: LeptonXAbpCoreModule, declarations: [IfReplaceableTemplateExistsDirective], imports: [CommonModule, PageAlertContainerModule], exports: [IfReplaceableTemplateExistsDirective, PageAlertContainerModule] });
LeptonXAbpCoreModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: LeptonXAbpCoreModule, imports: [CommonModule, PageAlertContainerModule, PageAlertContainerModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: LeptonXAbpCoreModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, PageAlertContainerModule],
                    declarations: [IfReplaceableTemplateExistsDirective],
                    exports: [IfReplaceableTemplateExistsDirective, PageAlertContainerModule],
                }]
        }] });
