// @volo/ngx-lepton-x.core/esm2020/lib/components/footer/footer.module.mjs
import { NgModule } from '@angular/core';
import { FooterComponent } from './footer.component';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';
import * as i0 from "@angular/core";
export class LpxFooterModule {
    static forRoot() {
        return {
            ngModule: LpxFooterModule,
            providers: [],
        };
    }
}
LpxFooterModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: LpxFooterModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
LpxFooterModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.2.0", ngImport: i0, type: LpxFooterModule, declarations: [FooterComponent], imports: [CommonModule, RouterModule], exports: [FooterComponent] });
LpxFooterModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: LpxFooterModule, imports: [CommonModule, RouterModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: LpxFooterModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [FooterComponent],
                    exports: [FooterComponent],
                    imports: [CommonModule, RouterModule],
                }]
        }] });
