// @volo/ngx-lepton-x.lite/esm2020/lib/components/language-selection/language-selection.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { NgbDropdownModule } from '@ng-bootstrap/ng-bootstrap';
import { LpxIconModule } from '@volo/ngx-lepton-x.core';
import { LanguageSelectionComponent } from './language-selection.component';
import * as i0 from "@angular/core";
export class LanguageSelectionModule {
}
LanguageSelectionModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: LanguageSelectionModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
LanguageSelectionModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.2.0", ngImport: i0, type: LanguageSelectionModule, declarations: [LanguageSelectionComponent], imports: [CommonModule, NgbDropdownModule, LpxIconModule], exports: [LanguageSelectionComponent] });
LanguageSelectionModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: LanguageSelectionModule, imports: [CommonModule, NgbDropdownModule, LpxIconModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: LanguageSelectionModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [LanguageSelectionComponent],
                    imports: [CommonModule, NgbDropdownModule, LpxIconModule],
                    exports: [LanguageSelectionComponent],
                }]
        }] });
