// @volo/ngx-lepton-x.core/esm2020/lib/panels/panels.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { BreadcrumbPanelDirective } from './breadcrumb-panel.directive';
import { NavbarPanelDirective } from './navbar-panel.directive';
import { FooterPanelDirective } from './footer-panel.directive';
import { MobileNavbarPanelDirective } from './mobile-navbar-panel.directive';
import { ToolbarPanelDirective } from './toolbar-panel.directive';
import { TopNavbarPanelDirective } from './top-navbar-panel.directive';
import { SettingsPanelDirective } from './settings-panel.directive';
import * as i0 from "@angular/core";
const declarationsAndExports = [
    BreadcrumbPanelDirective,
    NavbarPanelDirective,
    FooterPanelDirective,
    MobileNavbarPanelDirective,
    ToolbarPanelDirective,
    TopNavbarPanelDirective,
    SettingsPanelDirective,
];
export class PanelsModule {
}
PanelsModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: PanelsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
PanelsModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.2.0", ngImport: i0, type: PanelsModule, declarations: [BreadcrumbPanelDirective,
        NavbarPanelDirective,
        FooterPanelDirective,
        MobileNavbarPanelDirective,
        ToolbarPanelDirective,
        TopNavbarPanelDirective,
        SettingsPanelDirective], imports: [CommonModule], exports: [BreadcrumbPanelDirective,
        NavbarPanelDirective,
        FooterPanelDirective,
        MobileNavbarPanelDirective,
        ToolbarPanelDirective,
        TopNavbarPanelDirective,
        SettingsPanelDirective] });
PanelsModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: PanelsModule, imports: [CommonModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: PanelsModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [...declarationsAndExports],
                    imports: [CommonModule],
                    exports: [...declarationsAndExports],
                }]
        }] });
