// @volo/ngx-lepton-x.core/esm2020/lib/services/language/language.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { LPX_LANGUAGE } from './tokens';
import { LanguageService } from './language.service';
import { LanguageTranslateDefaults } from './language-translate-keys.enum';
import { LPX_TRANSLATE_TOKEN } from '../translate/tokens';
import * as i0 from "@angular/core";
export class LpxLanguageModule {
    static forRoot(options) {
        return {
            ngModule: LpxLanguageModule,
            providers: [
                {
                    provide: LPX_LANGUAGE,
                    useValue: options?.languages || [],
                },
                {
                    provide: LPX_TRANSLATE_TOKEN,
                    useValue: [LanguageTranslateDefaults],
                    multi: true,
                },
                LanguageService,
            ],
        };
    }
}
LpxLanguageModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: LpxLanguageModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
LpxLanguageModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.2.0", ngImport: i0, type: LpxLanguageModule, imports: [CommonModule] });
LpxLanguageModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: LpxLanguageModule, imports: [CommonModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: LpxLanguageModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [],
                    imports: [CommonModule],
                }]
        }] });
