// @volo/abp.ng.lepton-x.core/esm2020/lib/services/document-dir.service.mjs
import { Inject, Injectable } from '@angular/core';
import { DocumentDirHandlerService } from '@abp/ng.theme.shared';
import { DOCUMENT } from '@angular/common';
import * as i0 from "@angular/core";
import * as i1 from "@abp/ng.theme.shared";
export class DocumentDirService {
    constructor(documentDirHandler, document) {
        this.documentDirHandler = documentDirHandler;
        this.document = document;
    }
    listenDir() {
        this.documentDirHandler.dir$.subscribe((dir) => {
            this.document.documentElement.dir = dir;
        });
    }
}
DocumentDirService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: DocumentDirService, deps: [{ token: i1.DocumentDirHandlerService }, { token: DOCUMENT }], target: i0.ɵɵFactoryTarget.Injectable });
DocumentDirService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: DocumentDirService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: DocumentDirService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.DocumentDirHandlerService }, { type: Document, decorators: [{
                    type: Inject,
                    args: [DOCUMENT]
                }] }]; } });
