// @abp/ng.theme.lepton-x/esm2020/lib/components/validation-error/validation-error.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ValidationErrorComponent } from './validation-error.component';
import { NgxValidateCoreModule, VALIDATION_ERROR_TEMPLATE, VALIDATION_INVALID_CLASSES, VALIDATION_TARGET_SELECTOR, } from '@ngx-validate/core';
import { CoreModule } from '@abp/ng.core';
import * as i0 from "@angular/core";
export class ValidationErrorModule {
    static forRoot() {
        return {
            ngModule: ValidationErrorModule,
            providers: [
                {
                    provide: VALIDATION_ERROR_TEMPLATE,
                    useValue: ValidationErrorComponent,
                },
                {
                    provide: VALIDATION_TARGET_SELECTOR,
                    useValue: '*:has(> .form-control)',
                },
                {
                    provide: VALIDATION_INVALID_CLASSES,
                    useValue: 'is-invalid',
                },
            ],
        };
    }
}
ValidationErrorModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: ValidationErrorModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ValidationErrorModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.2.0", ngImport: i0, type: ValidationErrorModule, declarations: [ValidationErrorComponent], imports: [CommonModule, CoreModule, NgxValidateCoreModule], exports: [ValidationErrorComponent, NgxValidateCoreModule] });
ValidationErrorModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: ValidationErrorModule, imports: [CommonModule, CoreModule, NgxValidateCoreModule, NgxValidateCoreModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: ValidationErrorModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [ValidationErrorComponent],
                    imports: [CommonModule, CoreModule, NgxValidateCoreModule],
                    exports: [ValidationErrorComponent, NgxValidateCoreModule],
                }]
        }] });
