// @abp/ng.theme.lepton-x/esm2020/layouts/side-menu-layout/providers/nav-item.provider.mjs
import { APP_INITIALIZER, Injector } from '@angular/core';
import { NavItemsService } from '@abp/ng.theme.shared';
import { LanguageSelectionComponent } from '../components/language-selection/language-selection.component';
import { UserProfileComponent } from '../components/user-profile/user-profile.component';
import { OAuthService } from 'angular-oauth2-oidc';
import { NavigateToLoginComponent } from '../components';
export const NAV_ITEM_PROVIDER = {
    provide: APP_INITIALIZER,
    multi: true,
    useFactory: addNavItems,
    deps: [Injector],
};
export function addNavItems(injector) {
    return () => {
        const navItems = injector.get(NavItemsService);
        const oAuthService = injector.get(OAuthService);
        navItems.addItems([
            {
                id: "Theme.LoginComponent" /* eThemeLeptonXComponents.Login */,
                order: 100,
                visible: () => !oAuthService.hasValidAccessToken(),
                component: NavigateToLoginComponent,
            },
            {
                id: "Theme.LanguagesComponent" /* eThemeLeptonXComponents.Languages */,
                order: 100,
                component: LanguageSelectionComponent,
            },
            {
                id: "Theme.CurrentUserComponent" /* eThemeLeptonXComponents.CurrentUser */,
                order: 100,
                component: UserProfileComponent,
            },
        ]);
    };
}
