// @volo/ngx-lepton-x.core/esm2020/lib/style/style.service.mjs
import { Inject, Injectable } from '@angular/core';
import { LPX_STYLE_FINAL } from './tokens';
import { DOCUMENT } from '@angular/common';
import { BehaviorSubject } from 'rxjs';
import * as i0 from "@angular/core";
export class StyleService {
    constructor(initialStyles, document) {
        this.initialStyles = initialStyles;
        this.document = document;
        this.lastInjectedStyle = null;
        this.initialized$ = new BehaviorSubject(false);
    }
    async initStyles(direction) {
        for (const style of this.initialStyles) {
            await this.loadStyle(style, direction);
        }
        this.initialized$.next(true);
    }
    async loadStyle(style, direction) {
        return new Promise((resolve, reject) => {
            const linkElem = this.createLinkElem(style, direction, resolve);
            //TODO: find a better way for understand style laaded by angular json
            const appStyles = document.querySelector('link[rel="stylesheet"][href*="styles"]');
            if (appStyles) {
                if (this.lastInjectedStyle && this.lastInjectedStyle.isConnected) {
                    this.lastInjectedStyle.insertAdjacentElement('afterend', linkElem);
                }
                else {
                    appStyles.insertAdjacentElement('beforebegin', linkElem);
                }
            }
            else {
                this.document.head.appendChild(linkElem);
            }
            this.lastInjectedStyle = linkElem;
            return Promise.resolve(linkElem);
        });
    }
    async replaceStyle(style, direction) {
        const loaded = this.document.querySelector(`link#${style.bundleName}`);
        if (loaded) {
            loaded.remove();
        }
        return this.loadStyle(style, direction);
    }
    async reloadInitialStyles(direction) {
        for (const style of this.initialStyles) {
            await this.replaceStyle(style, direction);
        }
    }
    createLinkElem(style, direction, resolve) {
        const linkElem = document.createElement('link');
        linkElem.rel = 'stylesheet';
        linkElem.id = style.bundleName;
        linkElem.href = `${style.bundleName}${direction === 'rtl' ? '.rtl' : ''}.css`;
        linkElem.onload = () => {
            resolve(linkElem);
        };
        return linkElem;
    }
}
StyleService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: StyleService, deps: [{ token: LPX_STYLE_FINAL }, { token: DOCUMENT }], target: i0.ɵɵFactoryTarget.Injectable });
StyleService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: StyleService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: StyleService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: undefined, decorators: [{
                    type: Inject,
                    args: [LPX_STYLE_FINAL]
                }] }, { type: Document, decorators: [{
                    type: Inject,
                    args: [DOCUMENT]
                }] }]; } });
