// @volo/ngx-lepton-x.core/esm2020/lib/pipes/translate/translate.pipe.mjs
import { Pipe } from '@angular/core';
import { LpxThemeTranslateService } from '../../services';
import * as i0 from "@angular/core";
import * as i1 from "../../services";
export class TranslatePipe {
    constructor(lpxThemeTranslateService) {
        this.lpxThemeTranslateService = lpxThemeTranslateService;
    }
    transform(value, ...args) {
        return this.lpxThemeTranslateService.translate$(value, args);
    }
}
TranslatePipe.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: TranslatePipe, deps: [{ token: i1.LpxThemeTranslateService }], target: i0.ɵɵFactoryTarget.Pipe });
TranslatePipe.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "14.0.0", version: "14.2.0", ngImport: i0, type: TranslatePipe, name: "lpxTranslate" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: TranslatePipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'lpxTranslate',
                }]
        }], ctorParameters: function () { return [{ type: i1.LpxThemeTranslateService }]; } });
