// ng-observe/esm2015/lib/ng-observe.js
import { ChangeDetectorRef, Inject, Injectable, InjectionToken, NgZone, Optional, } from '@angular/core';
import { isObservable, Subscription } from 'rxjs';
export const HASH_FN = new InjectionToken('HASH_FN', {
    providedIn: 'root',
    factory: createHashFn,
});
const BRAND = '__ngObserve__';
// @dynamic
export class ObserveService {
    constructor(cdRef, hash, zone) {
        this.cdRef = cdRef;
        this.hash = hash;
        this.hooks = new Map();
        this.detectChanges = () => this.cdRef.detectChanges();
        this.collection = (sources, options = {}) => {
            const sink = Array.isArray(sources) ? [] : {};
            Object.defineProperty(sink, BRAND, {
                value: true,
                enumerable: false,
                writable: false,
            });
            const observe = this.observe(sink);
            Object.keys(sources).forEach(key => {
                const source = sources[key];
                const option = options[key];
                observe(key, source, option);
            });
            return sink;
        };
        this.value = (source, options) => {
            const sink = {};
            this.observe(sink)('value', source, options);
            return toValue(sink, 'value');
        };
        if (zone instanceof NgZone) {
            this.detectChanges = () => this.cdRef.markForCheck();
        }
    }
    createUniqueId(key) {
        try {
            throw new Error();
        }
        catch (e) {
            return String(this.hash(e.stack + String(key)));
        }
    }
    observe(sink) {
        const fn = (key, source, { uniqueId = this.createUniqueId(key), errorHandler = () => { } } = {}) => {
            let subscription = new Subscription();
            const noop = () => { };
            const unsubscribe = () => subscription.unsubscribe();
            const complete = () => {
                (this.hooks.get(uniqueId) || noop)();
                this.hooks.delete(uniqueId);
            };
            complete();
            this.hooks.set(uniqueId, unsubscribe);
            // tslint:disable-next-line: deprecation
            subscription = source.subscribe({
                next: x => {
                    sink[key] = x;
                    this.detectChanges();
                },
                error: errorHandler,
                complete,
            });
        };
        return fn;
    }
    ngOnDestroy() {
        this.hooks.forEach(unsubscribe => unsubscribe());
    }
}
ObserveService.decorators = [
    { type: Injectable }
];
ObserveService.ctorParameters = () => [
    { type: ChangeDetectorRef },
    { type: undefined, decorators: [{ type: Inject, args: [HASH_FN,] }] },
    { type: NgZone, decorators: [{ type: Optional }] }
];
export const OBSERVE = new InjectionToken('OBSERVE');
export const OBSERVE_PROVIDER = [
    ObserveService,
    {
        provide: OBSERVE,
        useFactory: observeFactory,
        deps: [ObserveService],
    },
];
export function observeFactory(service) {
    return (source, options) => isObservable(source)
        ? service.value(source, options)
        : service.collection(source, options);
}
export class Observed {
    constructor(seed, mapFn) {
        this.seed = seed;
        this.getter = () => mapFn(this.seed);
    }
    get value() {
        return this.getter();
    }
}
export function createHashFn() {
    const k = 2654435761;
    const shift = Math.imul ? (n) => Math.imul(n, k) : (n) => imul(n, k);
    const hashFn = (input) => {
        let index = input.length;
        let hash = 0xabadcafe;
        while (index--) {
            hash = shift(hash ^ input.charCodeAt(index));
        }
        return (hash ^ (hash >>> 16)) >>> 0;
    };
    return hashFn;
}
// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/imul
export function imul(a, b) {
    b |= 0;
    let result = (a & 0x003fffff) * b;
    if (a & 0xffc00000) {
        result += ((a & 0xffc00000) * b) | 0;
    }
    return result | 0;
}
export function isCollection(source) {
    return Boolean(source && source[BRAND]);
}
export function toMappedValue(collection, mapFn) {
    return new Observed(collection, mapFn);
}
export function toValue(collection, key) {
    return new Observed(collection, source => source[key]);
}
export function toValues(collection) {
    if (Array.isArray(collection)) {
        return collection.map((_, index) => new Observed(collection, source => source[index]));
    }
    const values = {};
    for (const key in collection) {
        if (collection.hasOwnProperty(key)) {
            values[key] = new Observed(collection, source => source[key]);
        }
    }
    return values;
}
