// @abp/ng.theme.lepton-x/esm2020/layouts/side-menu-layout/components/navigate-to-login/navigate-to-login.component.mjs
import { Component } from '@angular/core';
import { AuthService } from '@abp/ng.core';
import * as i0 from "@angular/core";
import * as i1 from "@abp/ng.core";
import * as i2 from "@angular/common";
import * as i3 from "@volo/ngx-lepton-x.core";
export class NavigateToLoginComponent {
    constructor(authService) {
        this.authService = authService;
    }
    navigateToLogin(event) {
        event.preventDefault();
        this.authService.navigateToLogin();
    }
}
NavigateToLoginComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: NavigateToLoginComponent, deps: [{ token: i1.AuthService }], target: i0.ɵɵFactoryTarget.Component });
NavigateToLoginComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "14.2.0", type: NavigateToLoginComponent, selector: "abp-navigate-to-login", ngImport: i0, template: `
    <a href="#" class="nav-link" (click)="navigateToLogin($event)">
      <span>{{ 'AbpUi::Login' | lpxTranslate | async }}</span>
    </a>
  `, isInline: true, dependencies: [{ kind: "pipe", type: i2.AsyncPipe, name: "async" }, { kind: "pipe", type: i3.TranslatePipe, name: "lpxTranslate" }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: NavigateToLoginComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'abp-navigate-to-login',
                    template: `
    <a href="#" class="nav-link" (click)="navigateToLogin($event)">
      <span>{{ 'AbpUi::Login' | lpxTranslate | async }}</span>
    </a>
  `,
                }]
        }], ctorParameters: function () { return [{ type: i1.AuthService }]; } });
