// @abp/ng.account/esm2020/lib/account.module.mjs
import { CoreModule, LazyModuleFactory } from '@abp/ng.core';
import { ThemeSharedModule } from '@abp/ng.theme.shared';
import { NgModule } from '@angular/core';
import { NgbDropdownModule } from '@ng-bootstrap/ng-bootstrap';
import { NgxValidateCoreModule } from '@ngx-validate/core';
import { AccountRoutingModule } from './account-routing.module';
import { ChangePasswordComponent } from './components/change-password/change-password.component';
import { LoginComponent } from './components/login/login.component';
import { ManageProfileComponent } from './components/manage-profile/manage-profile.component';
import { PersonalSettingsComponent } from './components/personal-settings/personal-settings.component';
import { RegisterComponent } from './components/register/register.component';
import { ACCOUNT_CONFIG_OPTIONS } from './tokens/config-options.token';
import { accountConfigOptionsFactory } from './utils/factory-utils';
import { AuthenticationFlowGuard } from './guards/authentication-flow.guard';
import { ForgotPasswordComponent } from './components/forgot-password/forgot-password.component';
import { ResetPasswordComponent } from './components/reset-password/reset-password.component';
import { RE_LOGIN_CONFIRMATION_TOKEN } from './tokens';
import { UiExtensionsModule } from '@abp/ng.theme.shared/extensions';
import { ACCOUNT_EDIT_FORM_PROP_CONTRIBUTORS } from './tokens/extensions.token';
import { AccountExtensionsGuard } from './guards/extensions.guard';
import { PersonalSettingsHalfRowComponent } from './components/personal-settings/personal-settings-half-row.component';
import * as i0 from "@angular/core";
const declarations = [
    LoginComponent,
    RegisterComponent,
    ChangePasswordComponent,
    ManageProfileComponent,
    PersonalSettingsComponent,
    ForgotPasswordComponent,
    ResetPasswordComponent,
    PersonalSettingsHalfRowComponent,
];
export class AccountModule {
    static forChild(options = {}) {
        return {
            ngModule: AccountModule,
            providers: [
                AuthenticationFlowGuard,
                { provide: ACCOUNT_CONFIG_OPTIONS, useValue: options },
                {
                    provide: 'ACCOUNT_OPTIONS',
                    useFactory: accountConfigOptionsFactory,
                    deps: [ACCOUNT_CONFIG_OPTIONS],
                },
                {
                    provide: RE_LOGIN_CONFIRMATION_TOKEN,
                    useValue: options.isPersonalSettingsChangedConfirmationActive ?? true,
                },
                {
                    provide: ACCOUNT_EDIT_FORM_PROP_CONTRIBUTORS,
                    useValue: options.editFormPropContributors,
                },
                AccountExtensionsGuard,
            ],
        };
    }
    static forLazy(options = {}) {
        return new LazyModuleFactory(AccountModule.forChild(options));
    }
}
AccountModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.3", ngImport: i0, type: AccountModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
AccountModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.0.3", ngImport: i0, type: AccountModule, declarations: [LoginComponent,
        RegisterComponent,
        ChangePasswordComponent,
        ManageProfileComponent,
        PersonalSettingsComponent,
        ForgotPasswordComponent,
        ResetPasswordComponent,
        PersonalSettingsHalfRowComponent], imports: [CoreModule,
        AccountRoutingModule,
        ThemeSharedModule,
        NgbDropdownModule,
        NgxValidateCoreModule,
        UiExtensionsModule], exports: [LoginComponent,
        RegisterComponent,
        ChangePasswordComponent,
        ManageProfileComponent,
        PersonalSettingsComponent,
        ForgotPasswordComponent,
        ResetPasswordComponent,
        PersonalSettingsHalfRowComponent] });
AccountModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.0.3", ngImport: i0, type: AccountModule, imports: [CoreModule,
        AccountRoutingModule,
        ThemeSharedModule,
        NgbDropdownModule,
        NgxValidateCoreModule,
        UiExtensionsModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.3", ngImport: i0, type: AccountModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [...declarations],
                    imports: [
                        CoreModule,
                        AccountRoutingModule,
                        ThemeSharedModule,
                        NgbDropdownModule,
                        NgxValidateCoreModule,
                        UiExtensionsModule,
                    ],
                    exports: [...declarations],
                }]
        }] });
