// @volo/ngx-lepton-x.lite/esm2020/layouts/lib/components/mobile-navbar/mobile-navbar.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { LpxBrandLogoModule, LpxIconModule, LpxNavbarModule, } from '@volo/ngx-lepton-x.core';
import { LanguageSelectionModule } from '@volo/ngx-lepton-x.lite';
import { MobileNavbarComponent } from './mobile-navbar.component';
import * as i0 from "@angular/core";
export class LpxMobileNavbarModule {
}
LpxMobileNavbarModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: LpxMobileNavbarModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
LpxMobileNavbarModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.2.0", ngImport: i0, type: LpxMobileNavbarModule, declarations: [MobileNavbarComponent], imports: [CommonModule,
        LpxBrandLogoModule,
        LpxIconModule,
        LanguageSelectionModule,
        LpxNavbarModule], exports: [MobileNavbarComponent] });
LpxMobileNavbarModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: LpxMobileNavbarModule, imports: [CommonModule,
        LpxBrandLogoModule,
        LpxIconModule,
        LanguageSelectionModule,
        LpxNavbarModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: LpxMobileNavbarModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [MobileNavbarComponent],
                    imports: [
                        CommonModule,
                        LpxBrandLogoModule,
                        LpxIconModule,
                        LanguageSelectionModule,
                        LpxNavbarModule,
                    ],
                    exports: [MobileNavbarComponent],
                }]
        }] });
