// @volo/ngx-lepton-x.lite/esm2020/layouts/lib/side-menu-layout/side-menu-layout.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { LpxBreadcrumbModule, LpxFooterModule, LpxNavbarModule, LpxResponsiveModule, PanelsModule, } from '@volo/ngx-lepton-x.core';
import { LanguageSelectionModule, UserProfileModule, } from '@volo/ngx-lepton-x.lite';
import { LpxMobileNavbarModule } from '../components/mobile-navbar/mobile-navbar.module';
import { SettingsPanelDirective } from './panels/settings-panel.directive';
import { SideMenuLayoutComponent } from './side-menu-layout.component';
import { LanguageSelectionPanelDirective } from './panels/language-selection-panel.directive';
import { UserProfilePanelDirective } from './panels/user-profile-panel.directive';
import { TopbarContentPanelDirective } from './panels/topbar-content-panel.directive';
import { getSideMenuLayoutStylesProvider } from './providers';
import { MnUserProfileModule } from '../components';
import { MobileUserProfilePanelDirective } from './panels/mobile-user-profile-panel.directive';
import { MnLanguageSelectionModule } from '../components/mn-language-selection';
import { MobileLanguageSelectionDirective } from './panels/mobile-language-selection.directive';
import * as i0 from "@angular/core";
const declarationsWithExports = [
    SideMenuLayoutComponent,
    SettingsPanelDirective,
    LanguageSelectionPanelDirective,
    TopbarContentPanelDirective,
    UserProfilePanelDirective,
    MobileUserProfilePanelDirective,
    MobileLanguageSelectionDirective,
];
export class LpxSideMenuLayoutModule {
    static forRoot(options) {
        return {
            ngModule: LpxSideMenuLayoutModule,
            providers: [getSideMenuLayoutStylesProvider(options?.styleFactory)],
        };
    }
}
LpxSideMenuLayoutModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: LpxSideMenuLayoutModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
LpxSideMenuLayoutModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.2.0", ngImport: i0, type: LpxSideMenuLayoutModule, declarations: [SideMenuLayoutComponent,
        SettingsPanelDirective,
        LanguageSelectionPanelDirective,
        TopbarContentPanelDirective,
        UserProfilePanelDirective,
        MobileUserProfilePanelDirective,
        MobileLanguageSelectionDirective], imports: [CommonModule,
        LpxNavbarModule,
        LpxBreadcrumbModule,
        LpxResponsiveModule,
        UserProfileModule,
        LanguageSelectionModule,
        LpxMobileNavbarModule,
        MnUserProfileModule,
        MnLanguageSelectionModule,
        LpxFooterModule,
        PanelsModule], exports: [SideMenuLayoutComponent,
        SettingsPanelDirective,
        LanguageSelectionPanelDirective,
        TopbarContentPanelDirective,
        UserProfilePanelDirective,
        MobileUserProfilePanelDirective,
        MobileLanguageSelectionDirective] });
LpxSideMenuLayoutModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: LpxSideMenuLayoutModule, imports: [CommonModule,
        LpxNavbarModule,
        LpxBreadcrumbModule,
        LpxResponsiveModule,
        UserProfileModule,
        LanguageSelectionModule,
        LpxMobileNavbarModule,
        MnUserProfileModule,
        MnLanguageSelectionModule,
        LpxFooterModule,
        PanelsModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: LpxSideMenuLayoutModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [...declarationsWithExports],
                    imports: [
                        CommonModule,
                        LpxNavbarModule,
                        LpxBreadcrumbModule,
                        LpxResponsiveModule,
                        UserProfileModule,
                        LanguageSelectionModule,
                        LpxMobileNavbarModule,
                        MnUserProfileModule,
                        MnLanguageSelectionModule,
                        LpxFooterModule,
                        PanelsModule,
                    ],
                    exports: [...declarationsWithExports],
                }]
        }] });
