// @volo/ngx-lepton-x.core/esm2020/lib/components/breadcrumb/breadcrumb-route-listener.service.mjs
import { Injectable } from '@angular/core';
import { NavigationEnd, Router } from '@angular/router';
import { NavbarService } from '../navbar/navbar.service';
import { BreadcrumbService } from './breadcrumb.service';
import { combineLatest } from 'rxjs';
import { filter } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "../navbar/navbar.service";
import * as i2 from "@angular/router";
import * as i3 from "./breadcrumb.service";
export class BreadcrumbRouteListenerService {
    constructor(navbarService, router, breadcrumbService) {
        this.navbarService = navbarService;
        this.router = router;
        this.breadcrumbService = breadcrumbService;
    }
    subscribeRoute() {
        combineLatest([
            this.router.events.pipe(filter((event) => event instanceof NavigationEnd)),
            this.navbarService.navbarItems$.pipe(filter((items) => !!items.length)),
        ]).subscribe(([event, items]) => {
            let activeItem = this.navbarService.findByLink(event.url);
            if (!activeItem.item) {
                activeItem = this.navbarService.findByLink('/');
            }
            const breadCrumbItems = activeItem.location.reduce((acc, itemIndex) => {
                const parent = acc[acc.length - 1]?.children || items;
                const item = parent[itemIndex];
                return [
                    ...acc,
                    { ...item, siblings: parent },
                ];
            }, []);
            this.breadcrumbService.setItems(this.mapNavbarItemToBreadcrumbItem(breadCrumbItems));
        });
    }
    mapNavbarItemToBreadcrumbItem(items) {
        return items.map(({ text, link, icon, siblings }) => ({
            text: text || '',
            link,
            icon,
            children: this.mapNavbarItemToBreadcrumbItem(siblings || []),
        }));
    }
}
BreadcrumbRouteListenerService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: BreadcrumbRouteListenerService, deps: [{ token: i1.NavbarService }, { token: i2.Router }, { token: i3.BreadcrumbService }], target: i0.ɵɵFactoryTarget.Injectable });
BreadcrumbRouteListenerService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: BreadcrumbRouteListenerService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: BreadcrumbRouteListenerService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.NavbarService }, { type: i2.Router }, { type: i3.BreadcrumbService }]; } });
