// @volo/ngx-lepton-x.core/esm2020/lib/components/breadcrumb/breadcrumb.module.mjs
import { APP_INITIALIZER, Injector, NgModule, } from '@angular/core';
import { CommonModule } from '@angular/common';
import { BreadcrumbComponent } from './breadcrumb.component';
import { LpxIconModule } from '../icon/icon.module';
import { RouterModule } from '@angular/router';
import { LpxClickOutsideModule } from '../../directives/click-outside/click-outside.module';
import { BreadcrumbRouteListenerService } from './breadcrumb-route-listener.service';
import { ToObservableModule } from '../../pipes/to-observable/to-observable.module';
import * as i0 from "@angular/core";
export const exportedDeclarations = [BreadcrumbComponent];
export class LpxBreadcrumbModule {
    static forRoot() {
        return {
            ngModule: LpxBreadcrumbModule,
            providers: [
                {
                    provide: APP_INITIALIZER,
                    useFactory: breadCrumbInit,
                    multi: true,
                    deps: [Injector],
                },
            ],
        };
    }
}
LpxBreadcrumbModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: LpxBreadcrumbModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
LpxBreadcrumbModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.2.0", ngImport: i0, type: LpxBreadcrumbModule, declarations: [BreadcrumbComponent], imports: [CommonModule,
        LpxIconModule,
        ToObservableModule,
        RouterModule,
        LpxClickOutsideModule], exports: [BreadcrumbComponent] });
LpxBreadcrumbModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: LpxBreadcrumbModule, imports: [CommonModule,
        LpxIconModule,
        ToObservableModule,
        RouterModule,
        LpxClickOutsideModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: LpxBreadcrumbModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [...exportedDeclarations],
                    imports: [
                        CommonModule,
                        LpxIconModule,
                        ToObservableModule,
                        RouterModule,
                        LpxClickOutsideModule,
                    ],
                    exports: [...exportedDeclarations],
                }]
        }] });
export function breadCrumbInit(injector) {
    const subs = () => {
        const service = injector.get(BreadcrumbRouteListenerService);
        service.subscribeRoute();
    };
    return subs;
}
