// @volo/ngx-lepton-x.core/esm2020/lib/pipes/to-observable/to-observable.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ToObservablePipe } from './to-observable.pipe';
import * as i0 from "@angular/core";
export class ToObservableModule {
}
ToObservableModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: ToObservableModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ToObservableModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.2.0", ngImport: i0, type: ToObservableModule, declarations: [ToObservablePipe], imports: [CommonModule], exports: [ToObservablePipe] });
ToObservableModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: ToObservableModule, imports: [CommonModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.0", ngImport: i0, type: ToObservableModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [ToObservablePipe],
                    imports: [CommonModule],
                    exports: [ToObservablePipe],
                }]
        }] });
