// @abp/ng.setting-management/esm2020/config/lib/providers/route.provider.mjs
import { noop, RoutesService } from '@abp/ng.core';
import { APP_INITIALIZER, inject, InjectionToken } from '@angular/core';
import { debounceTime, map } from 'rxjs/operators';
import { SettingTabsService } from '../services/settings-tabs.service';
export function configureRoutes(routesService) {
    return () => {
        routesService.add([
            {
                name: "AbpSettingManagement::Settings" /* eSettingManagementRouteNames.Settings */,
                path: '/setting-management',
                parentName: "AbpUiNavigation::Menu:Administration" /* eThemeSharedRouteNames.Administration */,
                layout: "application" /* eLayoutType.application */,
                order: 100,
                iconClass: 'fa fa-cog',
            },
        ]);
    };
}
export const SETTING_MANAGEMENT_HAS_SETTING = new InjectionToken('SETTING_MANAGEMENT_HAS_SETTING', {
    factory: () => {
        const settingTabsService = inject(SettingTabsService);
        return settingTabsService.visible$.pipe(debounceTime(0), map(nodes => !!nodes.length));
    },
});
export const SETTING_MANAGEMENT_ROUTE_PROVIDERS = [
    { provide: APP_INITIALIZER, useFactory: configureRoutes, deps: [RoutesService], multi: true },
    {
        provide: APP_INITIALIZER,
        useFactory: noop,
        deps: [SETTING_MANAGEMENT_HAS_SETTING],
        multi: true,
    },
];
